#Phenotype-genotype divergence across a steep ecological gradient in Rana temporaria

Methods to follow: 

1. Whitlock & Guillaume 2009
2. Santiso et al. 2015 (Whitlock co-author)
3. 


##Aim: 




Judith has sampled from the following populations:

populations used in 2014


Site	|name	|Kanton	|no.	|Lat.	|Long.	|Swiss.X	|Swiss.Y	|Elev.| HiSeq plate | nr indivs
:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:
egel	|Egelsee|	SG	|SG 248	|47.25100	|9.49537|	755.679	|235.383|	445| H10 + H11|10
vilt	|Kiessammler Vilters|	SG	|SG 346|	47.03213	|9.44647|	752.601	|210.962|486| H12 + H13 |
siec|	Siechenstuden	|GR	|GR 386	|46.98947	|9.55116|	760.683|	206.430	|517| H12 + H13 |
munt|	Munte bei Cazis|	GR	|none|	46.72779|	9.43862	|752.867|	177.124	|648| H12 + H13 |
csee	|Chapfensee|	SG|	none	|47.04729	|9.38625|	747.983	|212.531|	1033 |H02 |6
shwe|	Schwendiseen|	SG	|none|	47.18790	|9.32981	|743.319	|228.540|	1159 |H01 + H02|
prad|	Pradischier|	GR|	none|	46.78438|	9.52786	|759.520	|183.590|	1449|H12 + H13|
stls|	Stelsersee|	GR|	none	|46.96549|	9.75013	|775.890	|204.190	|1672| H01 |7 
gruu	|Gr端nsee Upper|	GR	|none	|46.85761|	9.78676	|779.370|	192.287	|2120 |H03 + H04|5
bnnp	|Berninapass|	GR|	none|	46.40662|	10.02762|	799.500|	142.757	|2342 | H09 |10
flue|	Fl端elapass|	GR	|none	|46.74765	|9.95187	|792.140|	180.460	|2388 | H10 + H11|
mctn|	Muot Cotschen|	GR	|none	|46.48154	|9.71700	|774.934|	150.340	|2542 | H10 + H11|3



populations used in 2015		

lab experiment								

Site	|name	|Kanton	|no.	|Lat.	|Long.	|Swiss.X	|Swiss.Y	|Elev.| HiSeq plate | nr indivs
:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:
vilt	|Kiessammler Vilters|	SG	|SG 346|	47.03213	|9.44647|	752.601	|210.962|	486| H12 + H13 |
siec|	Siechenstuden	|GR	|GR 386	|46.98947	|9.55116|	760.683|	206.430	|517| H12 + H13 |
munt|	Munte bei Cazis|	GR	|none|	46.72779|	9.43862	|752.867|	177.124	|648| H12 + H13 |
wise|	Wisenfurt|	SG|	SG 201	|47.19467	|9.47862|	754.574|	229.89	|445|H14 + H15|
fess|	Fessis Seeli NW	|GL|	none|	47.01989|	9.13650	|729.760	|209.450| 2184|H14 + H15|	
bnnp|	Berninapass	|GR|	none|	46.40662	|10.02762	|799.500	|142.757	|2342| H09 |
flue|	Fl端elapass	|GR|	none|	46.74765|	9.95187	|792.140	|180.460	|2388| H10 + H11|
mctn|	Muot Cotschen	|GR	|none	|46.48154|	9.71700	|774.934|	150.340	|2542| H10 + H11|
								
Field experiment

Site	|name	|Kanton	|no.	|Lat.	|Long.	|Swiss.X	|Swiss.Y	|Elev.| HiSeq plate | nr indivs
:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:
feer	|Feerbach	|SG	|SG344|	47.03452	|9.42398|	750.910	|211.210	|500 | - |NA
siec|	Siechenstuden	|GR|	GR 386	|46.98947|	9.55116	|760.683|	206.430|	517 |H12 + H13|
bnnp|	Berninapass	|GR|	none|	46.40662|	10.02762|	799.500	|142.757	|2342| H09 |
flue|	Fl端elapass	|GR|	none|	46.74765|	9.95187|	792.140	|180.460	|2388| H10 + H11|


And result from her data thus far: 

![alt_txt][Fig1]
[Fig1]:https://cloud.githubusercontent.com/assets/12142475/10883181/8ae51fce-8170-11e5-9e9a-4402f5ed2287.png

![alt_txt][Fig2]
[Fig2]:https://cloud.githubusercontent.com/assets/12142475/10883183/8ae78462-8170-11e5-812a-c16e73358e5d.png

![alt_txt][Fig3]
[Fig3]:https://cloud.githubusercontent.com/assets/12142475/10883180/8ae3db0a-8170-11e5-8995-5d4229d693f2.png


###Data exploration:

So, the first things I need to do: 

1. PyRAD for the populations that I have
2. Filter data: 
3. Fst between populations
4. IBD
5. Isolation by elevation
6. Heterozygosity & diversity measures
7. Inbreeding



##1. PyRAD

70 samples have already been processed with pyRAD with the previous batch of samples. 

I started a run with the remaining 110 samples on 5 Nov 2015. 

ended 10 Nov 2015, evening. 


###UPDATE 12 May 2016

I will check whether depth has an effect on nucleotide diversity in this dataset: (use scripts from https://github.com/alexjvr1/R.temp/blob/master/pyRADopt.md). 

I removed munt06 because of a low read count: 

![alt_txt][indiv.READS]
[indiv.READS]:https://cloud.githubusercontent.com/assets/12142475/15229978/53704c20-1848-11e6-874c-109a7f18edae.png


It's concerning that there is such a high correlation between nucleotide diversity and depth in this dataset: 

![alt_txt][FQ.plot1]
[FQ.plot1]:https://cloud.githubusercontent.com/assets/12142475/15229977/536fe99c-1848-11e6-829a-53c3806abe49.png


If I look at the data per population, it looks a bit better, though: 

The average nr of reads per population doesn't seem correlated with mean nucleotide diversity per population.


![alt_txt][reads.pop]
[reads.pop]:https://cloud.githubusercontent.com/assets/12142475/15230368/221d7d58-184a-11e6-8304-d42274eb9424.png

![alt_txt][ndiv.pop]
[ndiv.pop]:https://cloud.githubusercontent.com/assets/12142475/15230369/221faa1a-184a-11e6-8b28-52e3c5f978f0.png





After looking at pyRAD results, I will rerun pyRAD with may het set to 0.7. And downsample reads from each individual to better estimate the 

###Filtering with vcftools

Filter for min 0.5 genotyping rate and mac =3. 

```
vcftools --vcf FQc94d6m4p3.vcf --max-missing 0.5 --mac 3 --recode --recode-INFO-all --out s1.FQ.vcf

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf FQc94d6m4p3.vcf
	--recode-INFO-all
	--mac 3
	--max-missing 0.5
	--out s1.FQ.vcf
	--recode
	
Eighth Header entry should be INFO: INFO    
After filtering, kept 180 out of 180 Individuals
Outputting VCF file...
After filtering, kept 39540 out of a possible 971452 Sites
Run Time = 73.00 seconds
```



Create and imiss file and draw a histogram
```
vcftools --vcf s1.FQ.vcf.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

Most of the samples are missing <50% of the markers: 

![alt_txt][Fig4]
[Fig4]:https://cloud.githubusercontent.com/assets/12142475/11089944/4fbf4604-886e-11e5-9110-bb0031b03861.png


And remove all samples with >0.5 missing data
```
mawk '$5 > 0.5' out.imiss | cut -f1 > lowDP.indiv

wc -l lowDP.indiv

cat lowDP.indiv

csee_11.fq.trim
csee_12.fq.trim
csee_13.fq.trim
egel_20.fq.trim
fess_02.fq.trim
gruu_03.fq.trim
munt_05.fq.trim
munt_06.fq.trim
munt_07.fq.trim
shwe_02.fq.trim
stls_08.fq.trim
wise_03.fq.trim

vcftools --vcf s1.FQ.vcf.recode.vcf --remove lowDP.indiv --recode --recode-INFO-all --out FQ.imiss50.FINAL


Excluding individuals in 'exclude' list
After filtering, kept 168 out of 180 Individuals
Outputting VCF file...
After filtering, kept 39540 out of a possible 39540 Sites
Run Time = 7.00 seconds
```


##Basic statistics

###1. PCA

Sensitivity analysis to see the effect of the MAF on PCA

Create 3 datasets
```
vcftools --vcf FQfinal.recode.vcf --maf 0.1 --recode --recode-INFO-all --out FQMAF.1

vcftools --vcf FQfinal.recode.vcf --maf 0.25 --recode --recode-INFO-all --out FQMAF.25

vcftools --vcf FQfinal.recode.vcf --maf 0.3 --recode --recode-INFO-all --out FQMAF.3
```

MAF 0.10 = 17607 out of a possible 24489 Sites

MAF 0.25 = 8349 out of a possible 24489 Sites

MAF 0.30 = 6358 out of a possible 24489 Sites


And compare the PCAs between them: 

1. convert files to plink
```

```

2. transfer them to the mac

```

```
3. run the following analyses

```
##Basic stats

##PCA
library("ade4")
library("adegenet")
library("pegas")


CH.plink <- read.PLINK("CH436.plink.raw") ##reads a plink file in as genlight object

CH.plink #check that it is a genlight object

indNames(CH.plink)

CH436_pop.names <- read.table("~/phd_20150212/Analysis/ddRAD/CH530ddRAD/CH436_pop.names.csv", header=T, quote="\"") #read in the population names file
CH436_pop.names.factors <- as.factor(CH436_pop.names$PopID) #and convert to a factor
summary(CH436_pop.names)

pop(CH.plink) <- (CH436_pop.names.factors) #assign population names from a text file
pop(CH.plink)  ##and check that they are correct

temp <- table(unlist(other(CH.plink)))
barplot(temp,main="distribution of NoAlleles per locus",xlab="Number of Alleles",ylab="Number of sites",col=heat.colors(4))

myFreq <- glMean(CH.plink)
hist(myFreq, proba=T, col="gold", xlab="Allele Frequencies", main="Distribution of (2nd) allele frequencies")
temp <- density(myFreq)
lines(temp$x, temp$y,*1.8, lwd=3)

##Pop structure
pca1 <- glPca(CH.plink) ##This displays a barplot of the eigenvalues and asks the user for a number of retained principal components

scatter(pca1, posi="bottomright") #scatter plot of PC 1& 2
title("PCA of CH Rana temporaria, axes 1 and 2")

#NJ tree
library(ape)
tre <- nj(dist(as.matrix(CH.plink)))
tre

```

WC weighted pairwise Fst (MAF_0.25)

![alt_txt][Fig4]
[Fig4]:https://cloud.githubusercontent.com/assets/12142475/11095952/db4243c4-8897-11e5-848d-d9e7d71812db.png


Average weighted WC Fst for 

High (>1700 m): 0.194 

	- gruu
	- fess
	- bnnp
	- flue
	- mctn

Mid (901-1700m): 0.099

	- csee
	- shwe
	- prad
	- stls
	
Low (250-900m): 0.055

	- egel
	- wise
	- vilt
	- siec
	- munt

##Isolation by distance

- calculated great circle distance with R

```
#Great circle distance between samples

##Geographic distance between all sequenced R.temp CH samples
##using package Rdist

library(fields)
#Import .csv with coordinates

FstQst <- read.csv("FstQst_RAD_coords_20151111.csv", header=T)
FstQst$Long.



#rdist.earth (in fields package) wants only long & lat
CH_RAD_lon.lat <- cbind(FstQst$Long., FstQst$Lat.)
CH_RAD_lon.lat

#calculate great circle distances
distance.matrix.CHRAD <- rdist.earth(CH_RAD_lon.lat)
summary(distance.matrix.CHRAD)
dim(distance.matrix.CHRAD)

#and use only the lower half of the matrix
upper.tri(distance.matrix.CHRAD)
distance.matrix.CHRAD[lower.tri(distance.matrix.CHRAD)]<-NA
distance.matrix.CHRAD

#change from matrix to dataframe
bli <- as.data.frame(distance.matrix.CHRAD)
head(bli)
colnames(bli) <- FstQst$Site
rownames(bli) <- FstQst$Site

bli[lower.tri(bli,diag=TRUE)]=NA  #Prepare to drop duplicates and meaningless information
bli=as.data.frame(as.table(as.matrix(bli)))  #Turn into a 3-column table
bli
bli=na.omit(bli)  #Get rid of the junk we flagged above
bli
colnames(bli)<-c("site1", "site2", "dist(km)")
head(bli)

bli2 <- bli[sort("site2"),]

head(bli2)


##write to csv
write.csv(bli, file="distance.FstQst.csv",row.names=F)
```

Compare with pairwise Fst calculated above. 

Run IBD analysis with Genalex (excel plugin). - it's possible to specify that the data is in columns rather than tables.

plot in R
```
##IBD

IBD <- read.csv("FQ_IBDgraph.csv")
head(IBD)


library(ggplot2)
IBDplot <- ggplot(IBD, aes(x=dist, y=Fst)) + labs(x="distance (km)", y="Fst/1-Fst", title="Isolation by Distance") + geom_point(shape=1) + geom_smooth(method=lm, formula=y~x)

lm_eqn <- function(IBD){
  m <- lm(y ~ x, IBD);
  eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
                   list(a = format(coef(m)[1], digits = 2), 
                        b = format(coef(m)[2], digits = 2), 
                        r2 = format(summary(m)$r.squared, digits = 3)))
  as.character(as.expression(eq));                 
}

p1 <- IBDplot + geom_text(x = 50, y = 0.6, label = lm_eqn(IBD), parse = TRUE)
```


![alt_txt][Fig5]
[Fig5]:https://cloud.githubusercontent.com/assets/12142475/11118726/6825824a-8944-11e5-98f9-574d529c3369.png

y = 0.0035x - 0.025

R2 = 0.47597


## AMOVA



## PCA



## NJtree



## mtDNA lineages

There is a northern and southern CH lineage. The distribution for these populations is as follows: 


pop | long|lat|elevation | CHN | CHS | n
:--:|:--:|:--:|:--:|:--:|:--:|:--:
bnnp|46.40662|10.02762|2342|0|1|4
egel|	47.25100|	9.49537	|445|1|0|4
flue|	46.74765|	9.95187|	2388|0.5|0.5|4
gruu|	46.85761|	9.78676	|2120|1|0|4
mctn|	46.48154|	9.71700|	2542|0|1|4
prad|	46.78438|	9.52786|	1449|0.33|0.67|3
shwe|	47.18790|	9.32981|	1159|1|0|4
siec|	46.98947|	9.55116|	517|0.67|0.33|3
stls|	46.96549|	9.75013|	1672|0|1|4


##Diversity measures

nucleotide diversity vs elevation

He vs elevation

Fis vs elevation








