#Fine scale connectivity or Rana temporaria at different spatial scales


Question: what determines connectivity between populations across different spatial scales? 
- same environmental gradient
- Very different landscapes

Important paper on the use of Mantel tests in spatial analyses: Don't!

Legendre et al. 2015 (Mehods in Ecology and Evolution)

http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12425/epdf

I'm interested in the at-site and between-site variables that could determine connectivity between populations. 

###Part 1. Gravity Model

Connectivity will be determined by at (v) and between (c) site variables, rather than absolute distance. 

For between site, I hypothesise that topographic complexity with create increased resistance to geneflow. 
a) At site, I hypothesise that site quality i.t.o pond size, and climate i.t.o developmental degree days, will explain migration between ponds. 


###Part 2. Does this hold true at different spatial scales?

Compare Gravity model across all data vs within Structure groups. 

- CHall

- CHS
    CHS.TI
    CHS.VS
 
- CHN

- CZ
 
Major barriers to dispersal? Ridges? How will I define this? 
Monmoniers algorithm 

###Part 3. Can I identify the range edge? I hypothesis that the range edge is determined by limited developmental time for the tadpoles. ie. duration of snow cover (snow free days), or developmental degree days. 

This will be estimated by bottleneck signatures: 

excess heterozygosity

Fis

nucleotide diversity

H0: Range edge is associated with season length -> 

mean temp of laying date: lower = range edge (- correlation)

degree days: less = range edge (-correlation)

Shadow days: more = range edge 




#####Part 1: Gravity Model

As implemented in Murphy et al. 2010. 

They've just released an R package: GeNetIt

https://cran.r-project.org/web/packages/GeNetIt/GeNetIt.pdf

1. Build a network of all connections 

2. Test the best fit model (i.e. which variables best describe the genetic distance)

3. Compare to a smaller network (e.g. 10km - double the dispersal distance of R.temp)


Building the network in R
```
library(sp)

setwd("/Users/alexjvr/2016RADAnalysis/3_LandscapeGenetics/")

rtemp_sites <- read.csv("rtemp_sites.csv", header=T)

xy <- cbind(rtemp_site$lat, rtemp_site$long)  ##create a df with only the coordinates

pts <- SpatialPoints(xy)  ##turns them into spatial points for sp

Rtemp.Spatial <- SpatialPointsDataFrame(pts, rtemp_site)  ##create a SpatialPointsDF. This is needed to create the network below. 


###Creating the network
library(GeNetIt)

sat.graph.rtemp <- knn.graph(Rtemp.Spatial, row.names=Rtemp.Spatial@data[,"site"], long.lat=T) #saturated graph
graph.rtemp.10km <- knn.graph(Rtemp.Spatial, row.names=Rtemp.Spatial@data[,"site"], long.lat=T, max.dist=10)  ##long.lat=T for decimal degrees. 

##plot
par(mfrow=c(3,2)) ##creates box for two plots. c(3,2) means 3 rows, 2 columns

plot(sat.graph.rtemp.30km, col="grey")
points(Rtemp.Spatial, co="red", pch=20, cex=0.5)
box()
title("30km dist contrained")

plot(sat.graph.rtemp, col="grey")
points(Rtemp.Spatial, co="red", pch=20, cex=0.5)
box()
title("Saturated graph")

```

![alt_txt][Sat.graphs]
[Sat.graphs]:https://cloud.githubusercontent.com/assets/12142475/19343990/c140a4f2-9138-11e6-818f-c078e047b125.png



Between site landscape variables (c) have been obtained already by Josh. These are just along a line - measuring the proportion of different variables, and the nr of times crossing linear variables (e.g. roads, rivers, etc). 


At site variables (v) would include landscape and climate. 

Landscape was extracted for 4 buffers surrounding the ponds (0-200m, 201-500m, 501-1000, 1001-2000). Units = proportion of area covered. For linear objects, the km linear obj/km2 measured. 

I will calculate the correlation of each landscape variables within the different buffers: 

##Plan: 

Gravity model on each canton: 

This stays within the 30km distance limit for which Josh extracted data. And it makes sense according to the IBD plots. It seems like VS populations, all within 40km from each other, show no IBD pattern. 

1. Ticino

2. VS

3. Bern

4. EAST

This way I can compare between each canton, and also validate the model throughout Switzerland 

##Data

####response

Response variable

1. Fst

2. Euclidean distance


####w: distance between sites (calculated using great circle distance)

distance in km

####c: between site variables

1. - highways, railcrossings, rivers

2. streams, primary roads

3.1 elevational difference - are frogs more likely to move to similar elevations?

3.2. elev.releif - the topographic complexity between sites (m/m)

4. - urban - Josh showed this to be negative

5. - open field - Josh showed this to be negative

6. + forest - Josh showed this to be positive

7. - glaciers and rocks - conceptually the same. Likely negative. 

8. - open water

9. + wetland





####v: at site variables

#####climate

I've already selected the 6 variables based on correlation (see 5_CH_landscapeGenomics.md)


1.1. + sol.rad.60d - amount of solar radiation 60 days after eggs were laid. i.e. measure of temp related to egg development rate

1.2. + days.above.6 - Judith found 6deg to be significant for development. This is conceptually related to 1.1.

1.2. - shadow days  - this encompasses N/S facing slopes, and whether they're in shadow from surrounding peaks. This may be underestimated, especially at low elevation, due to canopy. 

2.1. - day 10cm - day at which there's less than 10cm of snow. ie. related to temp and shadow days (but <0.8 correlation)

2.2. - pcpt.60d - this means rain at low elevation, but could mean snow at high elevation. It is the most correlated to day.10cm, likely due to the high elevation sites. 

3. temp.laying.date - Temp of at estimated laying date. This may be reasonable at low elevation, but there may be a lot of variation in laying date and temp at high elevation. 


#####landscape

These are derived in 3 toruses around the site. 

t1 

t2 

t3 

I will run all models using these 3 levels seperately. 

1. - raillines, rivers, highways - these have been shown to negatively impact amphibian movement. I will group them together

2. - primary roads and streams: I will test these seperately and together

3. - trainstations & airports

4. - urban - these haven been shown to negatively impact amphibian movement

5. - rocks 

6. - glaciers

7. - open water

8. - fields - Josh showed that this negatively impacted movement

9. + wetland - I expect more amphibians in wetlands

10. + forest - not sure whether this will be good or bad. For movement it seems good, but may be bad for ponds. 

11. area

12. depth

13. canopy

14. vegetation



Correlation

```
library(corrplot)
library(caret)
setwd("/Users/alexjvr/2016RADAnalysis/3_LandscapeGenetics")

dat.climate <- read.table("Landscape.4.corrplot.txt", header=T)

torus1 <- cbind(dat.climate$urban.t1, dat.climate$field.t1, dat.climate$forest.t1, dat.climate$rocks.t1, dat.climate$water.t1, dat.climate$wetland.t1, dat.climate$highways.t1, dat.climate$primary.roads.t1, dat.climate$rail.t1, dat.climate$rivers.t1, dat.climate$streams.t1)  ##remove glaciers and stations since they're non-variable
x <- c("urban.t1", "field.t1", "forest.t1", "rocks.t1", "water.t1", "wetland.t1", "highways.t1", "primary.roads.t1", "rail.t1", "rivers.t1", "streams.t1")
colnames(torus1) <- x

torus1.scale <- scale(torus1, center=T, scale=T)
corMat.torus1 <- cor(torus1.scale)
corrplot(corMat.torus1, order="hclust", tl.cex=0.5)
highlyCor.torus1 <- findCorrelation(corMat.torus1, 0.8)
torus1.filtered.scale <- torus1.scale[, highlyCor.torus1]
corMat.torus1.filtered <- cor(torus1.filtered.scale)
par(mfrow=c(2,1))
title="torus1 corrplot"
corrplot(corMat.torus1, order="hclust", tl.cex=0.5)
corrplot(corMat.torus1.filtered, order="hclust", tl.cex=0.5)



torus2 <- cbind(dat.climate$urban.t2, dat.climate$stations.t2, dat.climate$field.t2, dat.climate$forest.t2, dat.climate$rocks.t2, dat.climate$water.t2, dat.climate$wetland.t2, dat.climate$highways.t2, dat.climate$primary.roads.t2, dat.climate$rail.t2, dat.climate$rivers.t2, dat.climate$streams.t2)
x <- c("urban.t2", "stations.t2", "field.t2", "forest.t2", "rocks.t2", "water.t2", "wetland.t2", "highways.t2", "primary.roads.t2", "rail.t2", "rivers.t2", "streams.t2")
colnames(torus2) <- x

torus2.scale <- scale(torus2, center=T, scale=T)
corMat.torus2 <- cor(torus2.scale)
corrplot(corMat.torus2, order="hclust", tl.cex=0.5)
highlyCor.torus2 <- findCorrelation(corMat.torus2, 0.8)
torus2.filtered.scale <- torus2.scale[, highlyCor.torus2]
corMat.torus2.filtered <- cor(torus2.filtered.scale)
par(mfrow=c(2,1))
title="torus2 corrplot"
corrplot(corMat.torus2, order="hclust", tl.cex=0.5)
corrplot(corMat.torus2.filtered, order="hclust", tl.cex=0.5)


torus3 <- cbind(dat.climate$urban.t3, dat.climate$stations.t3, dat.climate$field.t3, dat.climate$forest.t3, dat.climate$rocks.t3,dat.climate$glaciers.t3, dat.climate$water.t3, dat.climate$wetland.t3, dat.climate$highways.t3, dat.climate$primary.roads.t3, dat.climate$rail.t3, dat.climate$rivers.t3, dat.climate$streams.t3)
x <- c("urban.t3", "stations.t3", "field.t3", "forest.t3", "rocks.t3", "glaciers.t3", "water.t3", "wetland.t3", "highways.t3", "primary.roads.t3", "rail.t3", "rivers.t3", "streams.t3")
colnames(torus3) <- x

torus3.scale <- scale(torus3, center=T, scale=T)
corMat.torus3 <- cor(torus3.scale)
corrplot(corMat.torus3, order="hclust", tl.cex=0.5)
highlyCor.torus3 <- findCorrelation(corMat.torus3, 0.8)
torus3.filtered.scale <- torus3.scale[, highlyCor.torus3]
corMat.torus3.filtered <- cor(torus3.filtered.scale)
par(mfrow=c(2,1))
title="Torus3 corrplot"
corrplot(corMat.torus3, order="hclust", tl.cex=0.5)
corrplot(corMat.torus3.filtered, order="hclust", tl.cex=0.5)
```



Nothing filtered from these datasets

![alt_txt][torus1]
[torus1]: https://cloud.githubusercontent.com/assets/12142475/19725335/459fcdb2-9b85-11e6-82f1-f898fa497916.png

![alt_txt][torus2]
[torus2]:https://cloud.githubusercontent.com/assets/12142475/19725341/498539b2-9b85-11e6-8a5c-2999fb15ffce.png

![alt_txt][torus3]
[torus3]:https://cloud.githubusercontent.com/assets/12142475/19725120/1790d6d8-9b84-11e6-96b4-f6f84cb1cefe.png





##CZ





##Test Data1: Ticino

I will set up the protocol and test the method using Canton Ticino. 

16 populations

7-20 individuals per population

Total indivs: 177 (- zeni07 which looks like a CHN) 


Import Data1 CHall into R and subset to have only Ticino populations: 

```
library(adegenet)
library(hierfstat)
```

Calculate pairwise distances
```
library(fields)
CHS.TI_lon.lat <- cbind(CHS.TI.coords$long, CHS.TI.coords$lat)
distance.matrix.CHS.TI <- rdist.earth(CHS.TI_lon.lat, miles=F)

m.CHS.TI <- as.matrix(distance.matrix.CHS.TI)
rownames(m.CHS.TI) <- CHS.TI.coords$site
colnames(m.CHS.TI) <- CHS.TI.coords$site

CHS.TI.m2 <- melt(m.CHS.TI)
names(CHS.TI.m2) <- c("site1", "site2", "distance.km")

```


Calculate pairwise Fst  (Using Nei 1973 as implemented with adgenet pairwise.fst)
```
CHS.TI.Fst <- pairwise.fst(CHall.L3$CHS.TI, pop=NULL, res.type=c("dist", "matrix"))

m.CHS.TI.Fst <- as.matrix(CHS.TI.Fst)
rownames(m.CHS.TI.Fst) <- CHS.TI.coords$site
colnames(m.CHS.TI.Fst) <- CHS.TI.coords$site

m2.CHS.TI.Fst <- melt(m.CHS.TI.Fst)
names(m2.CHS.TI.Fst) <- c("site1", "site2", "Fst.Nei73")
```

Edward's distance based on gene frequencies (Euclidean). 
Euclidean distance = straight line/Direct. - in genetic distance this is an abstract distance (~PC) with no underlying biological assumptions.
```
CHS.TI.genpop <- genind2genpop(CHall$CHS.TI)

CHS.TI.Euc <- dist.genpop(CHall.genpop, method=2)

m.CHS.TI.Euc <- as.matrix(CHS.TI.Euc)
rownames(m.CHS.TI.Euc) <- CHS.TI.coords$site
colnames(m.CHS.TI.Euc) <- CHS.TI.coords$site

m2.CHS.TI.Euc <- melt(m.CHS.TI.Euc)
names(m2.CHS.TI.Euc) <- c("site1", "site2", "Euc.dist")

```

Join the 3 distances together: 
```
CHS.TI.3dist <- merge(m2.CHS.TI.Euc, m2.CHS.TI.Fst, by.x=c("site1", "site2"), by.y=c("site1", "site2"))
CHS.TI.3dist.2 <- merge(CHS.TI.3dist, CHS.TI.m2, by.x=c("site1", "site2"), by.y=c("site1", "site2"))

```


Add the climate data, between site variables, and at site variables


```
###Between site

CHS.env.data <- merge(CHS.env.data, CHS.TI.climate.Data, by.x="site2", by.y="site2")
```



###Part3: Range Edge

As a proxy for pop size, we can use the number of clutches that I counted at each pond. 

Compare this to the calculated Hs. 

```
##Read in data and create plots

Table1 <- read.table("Table1.sumstats", header=T)
muTable1$names <- factor(Table1$names, level=Table1$names)

q.Hs <- qplot(names, Hs, fill=factor((elev.level), levels=c("Low", "Mid", "High")), data=Table1, geom="boxplot") + scale_fill_manual(values=c("yellow","green","blue"))+theme_bw()+theme(legend.title=element_blank()) + theme(axis.text.x=element_blank(), axis.title.x=element_blank()) 

q.clutch <- qplot(names, Nr.clutches, fill=factor((elev.level), levels=c("Low", "Mid", "High")), data=Table1, geom="boxplot") + scale_fill_manual(values=c("yellow","green","blue"))+theme_bw()+theme(legend.title=element_blank()) + theme(axis.text.x=element_text(angle = 90, hjust = 0))

multiplot(q.clutch, q.Hs)

```

![alt_txt][Hs.clutch]
[Hs.clutch]:


```
###Function for multiplot in ggplot2
multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)

  # Make a list from the ... arguments and plotlist
  plots <- c(list(...), plotlist)

  numPlots = length(plots)

  # If layout is NULL, then use 'cols' to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                    ncol = cols, nrow = ceiling(numPlots/cols))
  }

 if (numPlots==1) {
    print(plots[[1]])

  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))

    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))

      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
```




