#CH: Adaptation to elevation

1. What are the signals of adaptation to elevation? 

2. Are they the same across different elevational gradients? 

3. Genomic turnover -> ie. where is the most selective pressure & what will the effects of future climate be?


##Data generation

For summary stats and pop structure, I need datasets with little missing data, 1 SNP per locus. 

1. Data filtering MAF, missingness, Het, 1 SNP per locus. 

Analyses: 

1. Pairwise Fst

2. Het per population

3. Fis per pop

4. nucleotide diversity per pop

5. PCA

##Datasets & Summary statistics

###1. Dataset for summary stats & pop structure

CHall.forPCA.recode.vcf

Analysis Plan: 

1. Summary statistics (nucleotide diversity, EHet & ObsHet, 

2. Fst between pops

3. Per pop inbreeding co-efficient (Fis)

4. Partitioning of variance in data - AMOVA
		- Within or between pops
		- CHN/CHS vs pops
		- elevation vs pops

5. IBD

6. Pop Structure
	- DAPC
	- fastStructure (hierarchical)
	- TESS3
	- Tree based on genetic distance


Based on these result, I will decide how to partition the data further. 

This dataset has little missing data. -> produced from s1 file as created below from the 23Gb .vcf start file. 

```
vcftools --vcf CHall_1027.s1.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out CHall.maxmiss0.8

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_1027.s1.recode.vcf
	--recode-INFO-all
	--max-missing 0.8
	--out CHall.maxmiss0.8
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 3368 out of a possible 142303 Sites
Run Time = 24.00 seconds

vcftools --vcf CHall.maxmiss0.8.recode.vcf --thin 200 --recode --recode-INFO-all --out CHall.maxmiss0.8_thin200

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.maxmiss0.8.recode.vcf
	--recode-INFO-all
	--thin 200
	--out CHall.maxmiss0.8_thin200
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 3368 Sites
Run Time = 2.00 seconds

vcftools --vcf CHall.maxmiss0.8_thin200.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][CHsubset_0.8]
[CHsubset_0.8]:https://cloud.githubusercontent.com/assets/12142475/18847292/eeefa0ae-8429-11e6-8a55-97543deee600.png

remove indivs with >0.5 missing data (from out.imiss)
```
vcftools --vcf CHall.maxmiss0.8_thin200.recode.vcf --remove 0.8remove --recode --recode-INFO-all --out CHall.forPCA

Parameters as interpreted:
	--vcf CHall.maxmiss0.8_thin200.recode.vcf
	--exclude 0.8remove
	--recode-INFO-all
	--out CHall.forPCA
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 1002 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 2.00 seconds
```

Remove duplicate of seeo_04 (seeo_04cat incuded with less missing data)

```
vcftools --vcf CHall.forPCA.recode.vcf --remove-indv seeo_04.fq.trim --recode --recode-INFO-all --out CHall.Dataset1
```

and rename the samples in the file
```

```

####Pop Structure

#####fastStructure

convert to fastStructure format using pgdSpider. In the spid file, specify fastStructure format, and choose SNPs as markers. Also provide the pop file. 

```
/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/fastStructure/CHall_Dataset1_1001.1842.str
```
Run fastStructure on the gdcserver. I had a lot of problems trying to install this on my computer!!

Again, run K1-10 x 10. These need to be specified manually

```
/usr/bin/python2.6 /usr/local/fastStructure-20150714/structure.py -K 1 --format=str --input=CHall_Dataset1_1001.1842.str --output=CHall.Data1_K1.1
```

How many Ks?
```
/usr/bin/python2.6 /usr/local/fastStructure-20150714/chooseK.py --input=CHall.Data1_K*

Model complexity that maximizes marginal likelihood = 8
Model components used to explain structure in data = 1
```






#####TESS3


Run this in the command line. 

First convert the .vcf to .geno using LEA in R. Make sure a later version of R is chosen.
```
source("http://bioconductor.org/biocLite.R")
biocLite("LEA")

library(LEA)

setwd("/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/TESS3/")
output = vcf2geno("CHall.Dataset1.Final_1001.1842.vcf")

	- number of detected individuals:	1001
	- number of detected loci:		1838

For SNP info, please check ./CHall.Dataset1.Final_1001.1842.vcfsnp.

4 line(s) were removed because these are not SNPs.
Please, check ./CHall.Dataset1.Final_1001.1842.removed file, for more informations.
```

TESS3 removes variants with >1 alt allele. The four that were removed had the notation: 

50266 9 . C A,T 20 PASS NS=924;DP=4 GT REMOVED
829281 29 . G A,T 20 PASS NS=882;DP=4 GT REMOVED
872771 2 . C A,G 20 PASS NS=1002;DP=4 GT REMOVED
1625311 29 . C T,G 20 PASS NS=947;DP=4 GT REMOVED


Now I need the .coords file, which is a file with a lat & long column for each individual (no individual names). 

```
nano coords.CHall.Dataset1
```

Copy TESS3 executable over to the current directory and run K1-10 x 10 itirations

```
cp ~/Applications/TESS3-master/build/TESS3 .


./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.1.Q -f K1.1.Fst -y K1.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.2.Q -f K1.2.Fst -y K1.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.3.Q -f K1.3.Fst -y K1.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.4.Q -f K1.4.Fst -y K1.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.5.Q -f K1.5.Fst -y K1.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.6.Q -f K1.6.Fst -y K1.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.7.Q -f K1.7.Fst -y K1.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.8.Q -f K1.8.Fst -y K1.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.9.Q -f K1.9.Fst -y K1.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.10.Q -f K1.10.Fst -y K1.10.sum -c 0.05

./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.1.Q -f K2.1.Fst -y K2.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.2.Q -f K2.2.Fst -y K2.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.3.Q -f K2.3.Fst -y K2.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.4.Q -f K2.4.Fst -y K2.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.5.Q -f K2.5.Fst -y K2.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.6.Q -f K2.6.Fst -y K2.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.7.Q -f K2.7.Fst -y K2.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.8.Q -f K2.8.Fst -y K2.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.9.Q -f K2.9.Fst -y K2.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.10.Q -f K2.10.Fst -y K2.10.sum -c 0.05

./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.1.Q -f K3.1.Fst -y K3.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.2.Q -f K3.2.Fst -y K3.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.3.Q -f K3.3.Fst -y K3.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.4.Q -f K3.4.Fst -y K3.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.5.Q -f K3.5.Fst -y K3.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.6.Q -f K3.6.Fst -y K3.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.7.Q -f K3.7.Fst -y K3.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.8.Q -f K3.8.Fst -y K3.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.9.Q -f K3.9.Fst -y K3.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.10.Q -f K3.10.Fst -y K3.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.1.Q -f K4.1.Fst -y K4.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.2.Q -f K4.2.Fst -y K4.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.3.Q -f K4.3.Fst -y K4.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.4.Q -f K4.4.Fst -y K4.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.5.Q -f K4.5.Fst -y K4.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.6.Q -f K4.6.Fst -y K4.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.7.Q -f K4.7.Fst -y K4.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.8.Q -f K4.8.Fst -y K4.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.9.Q -f K4.9.Fst -y K4.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.10.Q -f K4.10.Fst -y K4.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.1.Q -f K5.1.Fst -y K5.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.2.Q -f K5.2.Fst -y K5.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.3.Q -f K5.3.Fst -y K5.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.4.Q -f K5.4.Fst -y K5.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.5.Q -f K5.5.Fst -y K5.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.6.Q -f K5.6.Fst -y K5.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.7.Q -f K5.7.Fst -y K5.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.8.Q -f K5.8.Fst -y K5.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.9.Q -f K5.9.Fst -y K5.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.10.Q -f K5.10.Fst -y K5.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.1.Q -f K6.1.Fst -y K6.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.2.Q -f K6.2.Fst -y K6.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.3.Q -f K6.3.Fst -y K6.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.4.Q -f K6.4.Fst -y K6.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.5.Q -f K6.5.Fst -y K6.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.6.Q -f K6.6.Fst -y K6.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.7.Q -f K6.7.Fst -y K6.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.8.Q -f K6.8.Fst -y K6.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.9.Q -f K6.9.Fst -y K6.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.10.Q -f K6.10.Fst -y K6.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.1.Q -f K7.1.Fst -y K7.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.2.Q -f K7.2.Fst -y K7.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.3.Q -f K7.3.Fst -y K7.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.4.Q -f K7.4.Fst -y K7.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.5.Q -f K7.5.Fst -y K7.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.6.Q -f K7.6.Fst -y K7.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.7.Q -f K7.7.Fst -y K7.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.8.Q -f K7.8.Fst -y K7.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.9.Q -f K7.9.Fst -y K7.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.10.Q -f K7.10.Fst -y K7.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.1.Q -f K8.1.Fst -y K8.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.2.Q -f K8.2.Fst -y K8.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.3.Q -f K8.3.Fst -y K8.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.4.Q -f K8.4.Fst -y K8.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.5.Q -f K8.5.Fst -y K8.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.6.Q -f K8.6.Fst -y K8.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.7.Q -f K8.7.Fst -y K8.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.8.Q -f K8.8.Fst -y K8.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.9.Q -f K8.9.Fst -y K8.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.10.Q -f K8.10.Fst -y K8.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.1.Q -f K9.1.Fst -y K9.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.2.Q -f K9.2.Fst -y K9.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.3.Q -f K9.3.Fst -y K9.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.4.Q -f K9.4.Fst -y K9.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.5.Q -f K9.5.Fst -y K9.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.6.Q -f K9.6.Fst -y K9.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.7.Q -f K9.7.Fst -y K9.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.8.Q -f K9.8.Fst -y K9.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.9.Q -f K9.9.Fst -y K9.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.10.Q -f K9.10.Fst -y K9.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.1.Q -f K10.1.Fst -y K10.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.2.Q -f K10.2.Fst -y K10.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.3.Q -f K10.3.Fst -y K10.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.4.Q -f K10.4.Fst -y K10.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.5.Q -f K10.5.Fst -y K10.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.6.Q -f K10.6.Fst -y K10.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.7.Q -f K10.7.Fst -y K10.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.8.Q -f K10.8.Fst -y K10.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.9.Q -f K10.9.Fst -y K10.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.10.Q -f K10.10.Fst -y K10.10.sum -c 0.05
```

copy all the entropy scores to an excel sheet and plot in R

```



```


PCA to decide how to divide pops





#####PCAdapt


Use the Final .vcf input file. And divide the file according to the elevations. 
```
--vcf CHall.Dataset1.Final_1001.1842.vcf --keep HighLow.names --recode --recode-INFO-all --out CH.all.HighLow

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep HighLow.names
	--recode-INFO-all
	--out CH.all.HighLow
	--recode

Keeping individuals in 'keep' list
After filtering, kept 737 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds
vpn-89-206-116-38:PCAdapt alexjvr$ vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --keep HighMid.names --recode --recode-INFO-all --out CH.all.HighMid

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep HighMid.names
	--recode-INFO-all
	--out CH.all.HighMid
	--recode

Keeping individuals in 'keep' list
After filtering, kept 689 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds
vpn-89-206-116-38:PCAdapt alexjvr$ vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --keep LowMid.names --recode --recode-INFO-all --out CH.all.LowMid

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep LowMid.names
	--recode-INFO-all
	--out CH.all.LowMid
	--recode

Keeping individuals in 'keep' list
After filtering, kept 576 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds


```

Run PCAdapt in R on terminal 

```
setwd("/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/PCAdapt")
library(pcadapt)
path_to_High.Low <- "CH.all.HighLow.recode.vcf"

High.Low <- read.pcadapt(path_to_High.Low, type="vcf")

Summary:

        - input file      CH.all.HighLow.recode.vcf
        - output file     CH.all.HighLow.recode.pcadapt

	- number of individuals detected:	737
	- number of loci detected:		1838

For SNP info, please check CH.all.HighLow.recode.vcfsnp.

4 line(s) have been removed because these are not SNPs.
Please, check CH.all.HighLow.recode.removed file, for more information.

File has been sucessfully converted.

```

Check the nr of PCs

```
x <- pcadapt(High.Low,K=20,transpose=F)

Reading file CH.all.HighLow.recode.pcadapt...
Number of SNPs: 1838
Number of individuals: 737
Number of SNPs with minor allele frequency lower than 0.05 ignored: 284
124630 out of 1354606 missing data ignored.

poplist <- read.table("HighLow.pop")

```

Plot the PCA using population information
```


```



1. All CH

MAC of 60 ~3%  (what percentage should I use for the initial stats and then for the rest of the analyses??)

```
vcftools --vcf CH_6.100.vcf --mac 60 --recode --recode-INFO-all --out CHall_1027.s1 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CH_6.100.vcf
	--recode-INFO-all
	--mac 60
	--out CHall_2027.s1
	--recode

Eighth Header entry should be INFO: INFO    
After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 142303 out of a possible 5784222 Sites
Run Time = 966.00 seconds

input file was 23Gb. Output 564Mb

vcftools --vcf CHall_1027.s1.recode.vcf --max-missing 0.5 --recode --recode-INFO-all --out CHall_1027.s2

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s1.recode.vcf
	--recode-INFO-all
	--max-missing 0.5
	--out CHall_2027.s2
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 20113 out of a possible 142303 Sites
Run Time = 34.00 seconds


vcftools --vcf CHall_1027.s2.recode.vcf --thin 200 --recode --recode-INFO-all --out CHall.thin200.recode.vcf

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s2.recode.vcf
	--recode-INFO-all
	--thin 500
	--out CHall_1027.s3
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 20113 Sites
Run Time = 7.00 seconds


vcftools --vcf CHall.thin200.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][CHmissing]
[CHmissing]:https://cloud.githubusercontent.com/assets/12142475/18841980/cda53a14-8413-11e6-85f1-fb6a6dd793e5.png


Based on the number of individuals lost per population, I've decided to keep individuals with <0.45 missing data. This means there is a bit of missing data, but it I don't lose any of the populations. 


```
vcftools --vcf CHall.thin200.recode.vcf --remove indvs.missingmorethan0.55 --recode --recode-INFO-all --out CHall_1027.s4

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_1027.s3.recode.vcf
	--exclude indvs.missingmorethan0.55
	--recode-INFO-all
	--out CHall_1027.s4
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 949 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 9760 Sites
Run Time = 6.00 seconds
```

Rename the 949 individuals
```

bcftools reheader CHall.thin200.recode.vcf -s new.names -o CHall.949.9760.vcf
```

4.Filter for >0.6 obs Het

Based on my recent checks on the pyRAD data, I should also filter all SNPs with >0.6 observed Heterozygosity.

I will do this in R using the PLINK file.

Convert to plink

```
vcftools --vcf CHall.949.9760.vcf --plink --out CH.949.9760.plink
```

Calculate HWE for all loci in PLINK

```
plink --file CH.949.9760.plink --hardy
```

PLINK output plink.hwe has a very strange format - multiple spaces between columns - so I couldn't figure out how to cut a specific column using linux

I sorted everything in excel.

There are only 89 SNPs with O.Het >0.6 (i.e. 0.91%)

```
nano SNPsHWE.remove
```

Remove with plink

```
plink --file CH.949.9760.plink --exclude SNPsHWE.remove --recodeA --recode --out CHall.949.9608.plink
```

Final dataset:

949 individuals

9608 SNPs

0.699 Genotyping rate

Use pgdspider to convert .ped PLINK file to vcf. And keep plink and vcf files on mac:

/Users/alexjvr/2016RADAnalysis/SE.MS1/input.files/

And replace the headers (pgdspider doubles the names). 



2. CHN-E



3. CHN-W



4. CHS-VS



5. CHS-TI



6. CHN.2



7. CHS.2



F-statistics

Het/nucleotide diversity

IBD


##Population Structure

TESS3


fastStructure



##Environmental variables: PCA

Decide which environmental variables to use. This shoudl probably be base on the results from the Landscape genetic analysis. 


##Outlier Analysis

###1. PCAdapt



###2. BayeScan



##EAA

###1. BayEnv


###2. LFMM



##Comparison between the different datasets & different analyses (Venn Diagram)





##Genomic Turnover: Gradient Forest






##Genomic Turnover: GDM






##Other Analyses??
