#CH: Adaptation to elevation

1. What are the signals of adaptation to elevation? 

2. Are they the same across different elevational gradients? 

3. Genomic turnover -> ie. where is the most selective pressure & what will the effects of future climate be?


##Data generation

For summary stats and pop structure, I need datasets with little missing data, 1 SNP per locus. 

1. Data filtering MAF, missingness, Het, 1 SNP per locus. 

Analyses: 

1. Pairwise Fst

2. Het per population

3. Fis per pop

4. nucleotide diversity per pop

5. PCA

##Datasets & Summary statistics

PCA to decide how to divide pops

1. All CH

MAC of 60 ~3%  (what percentage should I use for the initial stats and then for the rest of the analyses??)

```
vcftools --vcf CH_6.100.vcf --mac 60 --recode --recode-INFO-all --out CHall_1027.s1 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CH_6.100.vcf
	--recode-INFO-all
	--mac 60
	--out CHall_2027.s1
	--recode

Eighth Header entry should be INFO: INFO    
After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 142303 out of a possible 5784222 Sites
Run Time = 966.00 seconds

input file was 23Gb. Output 564Mb

vcftools --vcf CHall_1027.s1.recode.vcf --max-missing 0.5 --recode --recode-INFO-all --out CHall_1027.s2

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s1.recode.vcf
	--recode-INFO-all
	--max-missing 0.5
	--out CHall_2027.s2
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 20113 out of a possible 142303 Sites
Run Time = 34.00 seconds


vcftools --vcf CHall_1027.s2.recode.vcf --thin 200 --recode --recode-INFO-all --out CHall.thin200.recode.vcf

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s2.recode.vcf
	--recode-INFO-all
	--thin 500
	--out CHall_1027.s3
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 20113 Sites
Run Time = 7.00 seconds


vcftools --vcf CHall.thin200.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][CHmissing]
[CHmissing]:https://cloud.githubusercontent.com/assets/12142475/18841980/cda53a14-8413-11e6-85f1-fb6a6dd793e5.png


Based on the number of individuals lost per population, I've decided to keep individuals with <0.45 missing data. This means there is a bit of missing data, but it I don't lose any of the populations. 


```
vcftools --vcf CHall.thin200.recode.vcf --remove indvs.missingmorethan0.55 --recode --recode-INFO-all --out CHall_1027.s4

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_1027.s3.recode.vcf
	--exclude indvs.missingmorethan0.55
	--recode-INFO-all
	--out CHall_1027.s4
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 949 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 9760 Sites
Run Time = 6.00 seconds
```

Rename the 949 individuals
```

bcftools reheader CHall.thin200.recode.vcf -s new.names -o CHall.949.9760.vcf
```

4.Filter for >0.6 obs Het

Based on my recent checks on the pyRAD data, I should also filter all SNPs with >0.6 observed Heterozygosity.

I will do this in R using the PLINK file.

Convert to plink

```
vcftools --vcf CHall.949.9760.vcf --plink --out CHall.plink_
```

Calculate HWE for all loci in PLINK

```
plink --file SEsubset.s4 --hardy
```

PLINK output plink.hwe has a very strange format - multiple spaces between columns - so I couldn't figure out how to cut a specific column using linux

I sorted everything in excel.

There are only 30 SNPs with O.Het >0.6 (i.e. 0.56%)

```
923:86
45299:91
131938:21
31269:23
15018:23
107525:44
80914:106
109021:9
53856:9
31387:44
88409:3
116529:47
17502:52
3086:06:00
33252:37
62935:14
121937:13
85675:18
132142:1
45819:46
75796:21
58278:85
130290:76
71033:77
95774:102
121298:22
67455:2
100258:5
113742:72
26:17:00
```

Remove with plink

```
nano SNPstoexclude.txt

plink --file SEsubset.s4 --exclude SNPtoexclude.txt --recodeA --recode --out SEsubset.Final
```

Final dataset:

167 individuals

5217 SNPs

0.706 Genotyping rate

Use pgdspider to convert .ped PLINK file to vcf. And keep plink and vcf files on mac:

/Users/alexjvr/2016RADAnalysis/SE.MS1/input.files/

And replace the headers (pgdspider doubles the names). 



2. CHN-E



3. CHN-W



4. CHS-VS



5. CHS-TI



6. CHN.2



7. CHS.2



F-statistics

Het/nucleotide diversity

IBD


##Population Structure

TESS3


fastStructure



##Environmental variables: PCA

Decide which environmental variables to use. This shoudl probably be base on the results from the Landscape genetic analysis. 


##Outlier Analysis

###1. PCAdapt



###2. BayeScan



##EAA

###1. BayEnv


###2. LFMM



##Comparison between the different datasets & different analyses (Venn Diagram)





##Genomic Turnover: Gradient Forest






##Genomic Turnover: GDM






##Other Analyses??
