#CH: Adaptation to elevation

1. What are the signals of adaptation to elevation? 

2. Are they the same across different elevational gradients? 

3. Genomic turnover -> ie. where is the most selective pressure & what will the effects of future climate be?


##Data generation

For summary stats and pop structure, I need datasets with little missing data, 1 SNP per locus. 

1. Data filtering MAF, missingness, Het, 1 SNP per locus. 

Analyses: 

1. Pairwise Fst

2. Het per population

3. Fis per pop

4. nucleotide diversity per pop

5. PCA

##Datasets & Summary statistics

###1. Dataset for summary stats & pop structure

CHall.forPCA.recode.vcf

Analysis Plan: 

1. Summary statistics (nucleotide diversity, EHet & ObsHet) 

2. Fst between pops

3. Per pop inbreeding co-efficient (Fis)

4. Partitioning of variance in data - AMOVA
		- Within or between pops
		- CHN/CHS vs pops
		- elevation vs pops

5. IBD: all
	CHN
	CHS

6. Pop Structure
	- DAPC
	- fastStructure (hierarchical)
	- TESS3
	- Tree based on genetic distance


Based on these result, I will decide how to partition the data further. 

This dataset has little missing data. -> produced from s1 file as created below from the 23Gb .vcf start file. 

```
vcftools --vcf CHall_1027.s1.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out CHall.maxmiss0.8

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_1027.s1.recode.vcf
	--recode-INFO-all
	--max-missing 0.8
	--out CHall.maxmiss0.8
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 3368 out of a possible 142303 Sites
Run Time = 24.00 seconds

vcftools --vcf CHall.maxmiss0.8.recode.vcf --thin 200 --recode --recode-INFO-all --out CHall.maxmiss0.8_thin200

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.maxmiss0.8.recode.vcf
	--recode-INFO-all
	--thin 200
	--out CHall.maxmiss0.8_thin200
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 3368 Sites
Run Time = 2.00 seconds

vcftools --vcf CHall.maxmiss0.8_thin200.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][CHsubset_0.8]
[CHsubset_0.8]:https://cloud.githubusercontent.com/assets/12142475/18847292/eeefa0ae-8429-11e6-8a55-97543deee600.png

remove indivs with >0.5 missing data (from out.imiss)
```
vcftools --vcf CHall.maxmiss0.8_thin200.recode.vcf --remove 0.8remove --recode --recode-INFO-all --out CHall.forPCA

Parameters as interpreted:
	--vcf CHall.maxmiss0.8_thin200.recode.vcf
	--exclude 0.8remove
	--recode-INFO-all
	--out CHall.forPCA
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 1002 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 2.00 seconds
```

Remove duplicate of seeo_04 (seeo_04cat incuded with less missing data)

```
vcftools --vcf CHall.forPCA.recode.vcf --remove-indv seeo_04.fq.trim --recode --recode-INFO-all --out CHall.Dataset1
```

and rename the samples in the file. Specify the old and new names. 
```
bcftools reheader CHall.Dataset1.recode.vcf -s new.names -o CHall.Dataset1.Final_1001.1842.vcf
```

####6. Pop Structure

1. DAPC

2. fastStructure (hierarchical)

3. TESS3

4. distance-based Tree

#####1. DAPC

I will remove the contaminated/potentially problematic individuals from the vcf file before starting
```
vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --remove contaminated.indivs --recode --recode-INFO-all --out CHall.Dataset1_992.1842
```

Convert to Plink.raw for Adegenet. (this needs to be done on the server because I ran out of memory on the mac)

```
vcftools --vcf CHall.Dataset1_992.1842.recode.vcf --plink --out CHall.Dataset1_992.1842.PLINK

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1_992.1842.recode.vcf
	--out CHall.Dataset1_992.1842.PLINK
	--plink

After filtering, kept 992 out of 992 Individuals
Writing PLINK PED and MAP files ... 
	PLINK: Only outputting biallelic loci.
Done.
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds

plink --file CHall.Dataset1_992.1842.PLINK --recodeA 

@----------------------------------------------------------@
|        PLINK!       |     v1.07      |   10/Aug/2009     |
|----------------------------------------------------------|
|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |
|----------------------------------------------------------|
|  For documentation, citation & bug-report instructions:  |
|        http://pngu.mgh.harvard.edu/purcell/plink/        |
@----------------------------------------------------------@

Web-based version check ( --noweb to skip )
Connecting to web...  OK, v1.07 is current

+++ PLINK 1.9 is now available! See above website for details +++ 

Writing this text to log file [ plink.log ]
Analysis started: Wed Oct  5 11:33:47 2016

Options in effect:
	--file CHall.Dataset1_992.1842.PLINK
	--recodeA

1838 (of 1838) markers to be included from [ CHall.Dataset1_992.1842.PLINK.map ]
Warning, found 992 individuals with ambiguous sex codes
Writing list of these individuals to [ plink.nosex ]
992 individuals read from [ CHall.Dataset1_992.1842.PLINK.ped ] 
0 individuals with nonmissing phenotypes
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
0 cases, 0 controls and 992 missing
0 males, 0 females, and 992 of unspecified sex
Before frequency and genotyping pruning, there are 1838 SNPs
992 founders and 0 non-founders found
Total genotyping rate in remaining individuals is 0.895725
0 SNPs failed missingness test ( GENO > 1 )
0 SNPs failed frequency test ( MAF < 0 )
After frequency and genotyping pruning, there are 1838 SNPs
After filtering, 0 cases, 0 controls and 992 missing
After filtering, 0 males, 0 females, and 992 of unspecified sex
Writing recoded file to [ plink.raw ] 
```




#####2. fastStructure

With Plink input: 

I used to convert this to fastStructure format in pgdspider, but to check the sample order it's better to use the PLINK input file. 

Convert to PLINK and then to bed. I had to run this on the gdcserver, because it was too big. 
```
vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --plink --out CHall.Dataset1.1001.1842.PLINK

 plink --file CHall.Dataset1.1001.1842.PLINK --out CHall.Dataset1.1001.1838 --make-bed 
```
plink read only 1838 loci

```
Options in effect:
	--file CHall.Dataset1.1001.1842.PLINK
	--make-bed

1838 (of 1838) markers to be included from [ CHall.Dataset1.1001.1842.PLINK.map ]
Warning, found 1001 individuals with ambiguous sex codes
Writing list of these individuals to [ plink.nosex ]
1001 individuals read from [ CHall.Dataset1.1001.1842.PLINK.ped ] 
0 individuals with nonmissing phenotypes
Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)
Missing phenotype value is also -9
0 cases, 0 controls and 1001 missing
0 males, 0 females, and 1001 of unspecified sex
Before frequency and genotyping pruning, there are 1838 SNPs
1001 founders and 0 non-founders found
Total genotyping rate in remaining individuals is 0.895889
0 SNPs failed missingness test ( GENO > 1 )
0 SNPs failed frequency test ( MAF < 0 )
After frequency and genotyping pruning, there are 1838 SNPs
After filtering, 0 cases, 0 controls and 1001 missing
After filtering, 0 males, 0 females, and 1001 of unspecified sex
Writing pedigree information to [ plink.fam ] 
Writing map (extended format) information to [ plink.bim ] 
Writing genotype bitfile to [ plink.bed ] 
Using (default) SNP-major mode

```


run Structure for K2 and K3
```
/usr/bin/python2.6 /usr/local/fastStructure-20150714/structure.py -K 2 --format=bed --input=CHall.Dataset1.1001.1838 --output=CHall.sampleorder
```

Now look at the output with the correct sample names, and compare to the output from the optimisation fastStructure runs (K1-10x10) shown below. 

The order seems to be the same as what I got from fastStructure input file. I think this is because I didn't specify populations. 




With fastStructure input (most of the runs): 
convert to fastStructure format using pgdSpider. In the spid file, specify fastStructure format, and choose SNPs as markers. Also provide the pop file. 

```
/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/fastStructure/CHall_Dataset1_1001.1842.str
```
Run fastStructure on the gdcserver. I had a lot of problems trying to install this on my computer!!

Again, run K1-10 x 10. These need to be specified manually

```
/usr/bin/python2.6 /usr/local/fastStructure-20150714/structure.py -K 1 --format=str --input=CHall_Dataset1_1001.1842.str --output=CHall.Data1_K1.1
```

How many Ks?
```
/usr/bin/python2.6 /usr/local/fastStructure-20150714/chooseK.py --input=CHall.Data1_K*

Model complexity that maximizes marginal likelihood = 8
Model components used to explain structure in data = 1
```

Summarise structure plots with CLUMPP

First, create a paramfile. This can be a copy of the example file. Specify Datatype 0 for individuals (vs 1 for pops), and number of individuals. Check the example from my run or from the example input for the chicken dataset. Most of the other CLUMPP parameters can be specified via command line, which will override the paramfile. 

indfiles need to be created: paste all the structure outputs for a specific K below each other. No headers. 

I've specified Greedy option 2

```
CLUMPP paramfile -k 2 -i K2.Q.indfile.txt -o K2.meanQ.out
```

For these runs, the sample order is the same between the PLINK test run and the input files that I created using pgdspider (i.e. fastStructure). 




Based on the Structure analysis, I've identified some possible contaminants: individuals that assign to populations that they should not assign to. They were all sequenced on plate H22. The rest of the individuals seem to be assigning correctly. 

gola_19 - looks right, but note in HiSeqPlates_20150723.xls shows it was contaminated by an SE sample  (plate H21)

H22: 

kebe_06 - 09

mart_08-10

ente_14





#####3. TESS3


Run this in the command line. 

First convert the .vcf to .geno using LEA in R. Make sure a later version of R is chosen.
```
source("http://bioconductor.org/biocLite.R")
biocLite("LEA")

library(LEA)

setwd("/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/TESS3/")
output = vcf2geno("CHall.Dataset1.Final_1001.1842.vcf")

	- number of detected individuals:	1001
	- number of detected loci:		1838

For SNP info, please check ./CHall.Dataset1.Final_1001.1842.vcfsnp.

4 line(s) were removed because these are not SNPs.
Please, check ./CHall.Dataset1.Final_1001.1842.removed file, for more informations.
```

TESS3 removes variants with >1 alt allele. The four that were removed had the notation: 

50266 9 . C A,T 20 PASS NS=924;DP=4 GT REMOVED
829281 29 . G A,T 20 PASS NS=882;DP=4 GT REMOVED
872771 2 . C A,G 20 PASS NS=1002;DP=4 GT REMOVED
1625311 29 . C T,G 20 PASS NS=947;DP=4 GT REMOVED


Now I need the .coords file, which is a file with a lat & long column for each individual (no individual names). 

```
nano coords.CHall.Dataset1
```

Copy TESS3 executable over to the current directory and run K1-10 x 10 itirations

```
cp ~/Applications/TESS3-master/build/TESS3 .


./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.1.Q -f K1.1.Fst -y K1.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.2.Q -f K1.2.Fst -y K1.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.3.Q -f K1.3.Fst -y K1.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.4.Q -f K1.4.Fst -y K1.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.5.Q -f K1.5.Fst -y K1.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.6.Q -f K1.6.Fst -y K1.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.7.Q -f K1.7.Fst -y K1.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.8.Q -f K1.8.Fst -y K1.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.9.Q -f K1.9.Fst -y K1.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 1 -q K1.10.Q -f K1.10.Fst -y K1.10.sum -c 0.05

./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.1.Q -f K2.1.Fst -y K2.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.2.Q -f K2.2.Fst -y K2.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.3.Q -f K2.3.Fst -y K2.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.4.Q -f K2.4.Fst -y K2.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.5.Q -f K2.5.Fst -y K2.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.6.Q -f K2.6.Fst -y K2.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.7.Q -f K2.7.Fst -y K2.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.8.Q -f K2.8.Fst -y K2.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.9.Q -f K2.9.Fst -y K2.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 2 -q K2.10.Q -f K2.10.Fst -y K2.10.sum -c 0.05

./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.1.Q -f K3.1.Fst -y K3.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.2.Q -f K3.2.Fst -y K3.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.3.Q -f K3.3.Fst -y K3.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.4.Q -f K3.4.Fst -y K3.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.5.Q -f K3.5.Fst -y K3.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.6.Q -f K3.6.Fst -y K3.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.7.Q -f K3.7.Fst -y K3.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.8.Q -f K3.8.Fst -y K3.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.9.Q -f K3.9.Fst -y K3.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 3 -q K3.10.Q -f K3.10.Fst -y K3.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.1.Q -f K4.1.Fst -y K4.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.2.Q -f K4.2.Fst -y K4.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.3.Q -f K4.3.Fst -y K4.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.4.Q -f K4.4.Fst -y K4.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.5.Q -f K4.5.Fst -y K4.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.6.Q -f K4.6.Fst -y K4.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.7.Q -f K4.7.Fst -y K4.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.8.Q -f K4.8.Fst -y K4.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.9.Q -f K4.9.Fst -y K4.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 4 -q K4.10.Q -f K4.10.Fst -y K4.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.1.Q -f K5.1.Fst -y K5.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.2.Q -f K5.2.Fst -y K5.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.3.Q -f K5.3.Fst -y K5.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.4.Q -f K5.4.Fst -y K5.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.5.Q -f K5.5.Fst -y K5.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.6.Q -f K5.6.Fst -y K5.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.7.Q -f K5.7.Fst -y K5.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.8.Q -f K5.8.Fst -y K5.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.9.Q -f K5.9.Fst -y K5.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 5 -q K5.10.Q -f K5.10.Fst -y K5.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.1.Q -f K6.1.Fst -y K6.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.2.Q -f K6.2.Fst -y K6.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.3.Q -f K6.3.Fst -y K6.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.4.Q -f K6.4.Fst -y K6.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.5.Q -f K6.5.Fst -y K6.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.6.Q -f K6.6.Fst -y K6.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.7.Q -f K6.7.Fst -y K6.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.8.Q -f K6.8.Fst -y K6.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.9.Q -f K6.9.Fst -y K6.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 6 -q K6.10.Q -f K6.10.Fst -y K6.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.1.Q -f K7.1.Fst -y K7.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.2.Q -f K7.2.Fst -y K7.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.3.Q -f K7.3.Fst -y K7.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.4.Q -f K7.4.Fst -y K7.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.5.Q -f K7.5.Fst -y K7.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.6.Q -f K7.6.Fst -y K7.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.7.Q -f K7.7.Fst -y K7.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.8.Q -f K7.8.Fst -y K7.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.9.Q -f K7.9.Fst -y K7.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 7 -q K7.10.Q -f K7.10.Fst -y K7.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.1.Q -f K8.1.Fst -y K8.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.2.Q -f K8.2.Fst -y K8.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.3.Q -f K8.3.Fst -y K8.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.4.Q -f K8.4.Fst -y K8.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.5.Q -f K8.5.Fst -y K8.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.6.Q -f K8.6.Fst -y K8.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.7.Q -f K8.7.Fst -y K8.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.8.Q -f K8.8.Fst -y K8.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.9.Q -f K8.9.Fst -y K8.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 8 -q K8.10.Q -f K8.10.Fst -y K8.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.1.Q -f K9.1.Fst -y K9.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.2.Q -f K9.2.Fst -y K9.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.3.Q -f K9.3.Fst -y K9.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.4.Q -f K9.4.Fst -y K9.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.5.Q -f K9.5.Fst -y K9.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.6.Q -f K9.6.Fst -y K9.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.7.Q -f K9.7.Fst -y K9.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.8.Q -f K9.8.Fst -y K9.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.9.Q -f K9.9.Fst -y K9.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 9 -q K9.10.Q -f K9.10.Fst -y K9.10.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.1.Q -f K10.1.Fst -y K10.1.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.2.Q -f K10.2.Fst -y K10.2.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.3.Q -f K10.3.Fst -y K10.3.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.4.Q -f K10.4.Fst -y K10.4.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.5.Q -f K10.5.Fst -y K10.5.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.6.Q -f K10.6.Fst -y K10.6.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.7.Q -f K10.7.Fst -y K10.7.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.8.Q -f K10.8.Fst -y K10.8.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.9.Q -f K10.9.Fst -y K10.9.sum -c 0.05
./TESS3 -x CHall.Dataset1.Final_1001.1842.geno -r coords.CH.Dataset1 -K 10-q K10.10.Q -f K10.10.Fst -y K10.10.sum -c 0.05
```

copy all the entropy scores to an excel sheet and plot in R

```



```

Use CLUMPP to average over all itirations of a given K. 

.indiv files need to be created by combining all the Q values into one file. And then paste 5 columns at the start of the doc as in Structure output. 
```
cat *2.Q > K2.allQ.indiv.csv
```

And manually paste the first five columns in. And leave a line open between runs. E.g. for K=2: 
```
1       1       (0)     1       :       0.576033        0.423967
2       2       (0)     1       :       0.855817        0.144183
3       3       (0)     1       :       0.787528        0.212472
4       4       (0)     1       :       0.827854        0.172146
5       5       (0)     1       :       0.822856        0.177144
6       6       (0)     1       :       0.806024        0.193976
7       7       (0)     1       :       0.913411        0.0865886
8       8       (0)     1       :       0.907361        0.0926394
9       9       (0)     1       :       0.797895        0.202105
10      10      (0)     1       :       0.849038        0.150962
```

copy these into the CLUMPP folder. Edit the .paramfile to reflect the number of runs. 
```
./CLUMPP Data1.paramfile -k 9 -i K9.allQ.txt -o K9.meanQ.out
```

These can be used to draw maps in R. 

But first choose the most likely K using the entropy scores copied from each run. 
```
###################################
######Graph of cross-entropy scores

library(ggplot2)

CHall.entropy <- read.csv("cross.Entropy.K1_10.csv")
CHall.entropy <- as.data.frame(CHall.entropy)
CHall.entropy

ggplot(CHall.entropy, aes(x=CHall.entropy$K, y=CHall.entropy$cross.entropy.CHD1))+geom_point(shape=1) + ggtitle("Cross-entropy for CHall_1001.1842 ") + ylab("Cross-entropy")+xlab("K")
```
![alt_txt][Cross.entropy]
[Cross.entropy]:https://cloud.githubusercontent.com/assets/12142475/19119730/00e7d26a-8b21-11e6-86f1-d5826fede84e.png



According to this, it looks like K=2 or K=4. 

K2-K4 in R: 

```
###Graphic display of TESS output
#########
library("fields")
library("RColorBrewer")
source("MapDisplay/POPSutilities.R")

Qmatrix <- read.table("K2.meanQ.out")
coords <- read.table("coords")
plot(coords, pch = 19, xlab = "Longitude", ylab= "Latitude")
#?map
map(add = T, boundary = T, interior = T, col = "grey80")

asc.raster=("srtm_38_02.asc")
asc.raster
grid=createGridFromAsciiRaster(asc.raster)  ##the .raster file needs to have YLLCENTER and XLLCENTER specified. Won't run otherwise. This can be changed manually using nano

constraints=getConstraintsFromAsciiRaster(asc.raster,cell_value_min=0)   ##constrains the map to the raster file size
maps(matrix = Qmatrix, coords, grid, method = "max", main = "Ancestry coefficients", xlab = "Longitude", ylab = "Latitude")
```


PCA to decide how to divide pops





#####PCAdapt


Use the Final .vcf input file. And divide the file according to the elevations. 
```
--vcf CHall.Dataset1.Final_1001.1842.vcf --keep HighLow.names --recode --recode-INFO-all --out CH.all.HighLow

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep HighLow.names
	--recode-INFO-all
	--out CH.all.HighLow
	--recode

Keeping individuals in 'keep' list
After filtering, kept 737 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds
vpn-89-206-116-38:PCAdapt alexjvr$ vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --keep HighMid.names --recode --recode-INFO-all --out CH.all.HighMid

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep HighMid.names
	--recode-INFO-all
	--out CH.all.HighMid
	--recode

Keeping individuals in 'keep' list
After filtering, kept 689 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds
vpn-89-206-116-38:PCAdapt alexjvr$ vcftools --vcf CHall.Dataset1.Final_1001.1842.vcf --keep LowMid.names --recode --recode-INFO-all --out CH.all.LowMid

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall.Dataset1.Final_1001.1842.vcf
	--keep LowMid.names
	--recode-INFO-all
	--out CH.all.LowMid
	--recode

Keeping individuals in 'keep' list
After filtering, kept 576 out of 1001 Individuals
Outputting VCF file...
After filtering, kept 1842 out of a possible 1842 Sites
Run Time = 1.00 seconds


```

Run PCAdapt in R on terminal 

```
setwd("/Users/alexjvr/2016RADAnalysis/5_CH.landscapeGenomics/Dataset1_forPCA/PCAdapt")
library(pcadapt)
path_to_High.Low <- "CH.all.HighLow.recode.vcf"

High.Low <- read.pcadapt(path_to_High.Low, type="vcf")

Summary:

        - input file      CH.all.HighLow.recode.vcf
        - output file     CH.all.HighLow.recode.pcadapt

	- number of individuals detected:	737
	- number of loci detected:		1838

For SNP info, please check CH.all.HighLow.recode.vcfsnp.

4 line(s) have been removed because these are not SNPs.
Please, check CH.all.HighLow.recode.removed file, for more information.

File has been sucessfully converted.

```

Check the nr of PCs

```
x <- pcadapt(High.Low,K=20,transpose=F)

Reading file CH.all.HighLow.recode.pcadapt...
Number of SNPs: 1838
Number of individuals: 737
Number of SNPs with minor allele frequency lower than 0.05 ignored: 284
124630 out of 1354606 missing data ignored.

poplist <- read.table("HighLow.pop")

```

Plot the PCA using population information
```


```



1. All CH

MAC of 60 ~3%  (what percentage should I use for the initial stats and then for the rest of the analyses??)

```
vcftools --vcf CH_6.100.vcf --mac 60 --recode --recode-INFO-all --out CHall_1027.s1 

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CH_6.100.vcf
	--recode-INFO-all
	--mac 60
	--out CHall_2027.s1
	--recode

Eighth Header entry should be INFO: INFO    
After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 142303 out of a possible 5784222 Sites
Run Time = 966.00 seconds

input file was 23Gb. Output 564Mb

vcftools --vcf CHall_1027.s1.recode.vcf --max-missing 0.5 --recode --recode-INFO-all --out CHall_1027.s2

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s1.recode.vcf
	--recode-INFO-all
	--max-missing 0.5
	--out CHall_2027.s2
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 20113 out of a possible 142303 Sites
Run Time = 34.00 seconds


vcftools --vcf CHall_1027.s2.recode.vcf --thin 200 --recode --recode-INFO-all --out CHall.thin200.recode.vcf

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_2027.s2.recode.vcf
	--recode-INFO-all
	--thin 500
	--out CHall_1027.s3
	--recode

After filtering, kept 1029 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 20113 Sites
Run Time = 7.00 seconds


vcftools --vcf CHall.thin200.recode.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][CHmissing]
[CHmissing]:https://cloud.githubusercontent.com/assets/12142475/18841980/cda53a14-8413-11e6-85f1-fb6a6dd793e5.png


Based on the number of individuals lost per population, I've decided to keep individuals with <0.45 missing data. This means there is a bit of missing data, but it I don't lose any of the populations. 


```
vcftools --vcf CHall.thin200.recode.vcf --remove indvs.missingmorethan0.55 --recode --recode-INFO-all --out CHall_1027.s4

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf CHall_1027.s3.recode.vcf
	--exclude indvs.missingmorethan0.55
	--recode-INFO-all
	--out CHall_1027.s4
	--recode

Excluding individuals in 'exclude' list
After filtering, kept 949 out of 1029 Individuals
Outputting VCF file...
After filtering, kept 9760 out of a possible 9760 Sites
Run Time = 6.00 seconds
```

Rename the 949 individuals
```

bcftools reheader CHall.thin200.recode.vcf -s new.names -o CHall.949.9760.vcf
```

4.Filter for >0.6 obs Het

Based on my recent checks on the pyRAD data, I should also filter all SNPs with >0.6 observed Heterozygosity.

I will do this in R using the PLINK file.

Convert to plink

```
vcftools --vcf CHall.949.9760.vcf --plink --out CH.949.9760.plink
```

Calculate HWE for all loci in PLINK

```
plink --file CH.949.9760.plink --hardy
```

PLINK output plink.hwe has a very strange format - multiple spaces between columns - so I couldn't figure out how to cut a specific column using linux

I sorted everything in excel.

There are only 89 SNPs with O.Het >0.6 (i.e. 0.91%)

```
nano SNPsHWE.remove
```

Remove with plink

```
plink --file CH.949.9760.plink --exclude SNPsHWE.remove --recodeA --recode --out CHall.949.9608.plink
```

Final dataset:

949 individuals

9608 SNPs

0.699 Genotyping rate

Use pgdspider to convert .ped PLINK file to vcf. And keep plink and vcf files on mac:

/Users/alexjvr/2016RADAnalysis/SE.MS1/input.files/

And replace the headers (pgdspider doubles the names). 



2. CHN-E



3. CHN-W



4. CHS-VS



5. CHS-TI



6. CHN.2



7. CHS.2



F-statistics

Het/nucleotide diversity

IBD


##Population Structure

TESS3


fastStructure



##Environmental variables: PCA

Decide which environmental variables to use. This shoudl probably be base on the results from the Landscape genetic analysis. 


##Outlier Analysis

###1. PCAdapt



###2. BayeScan



##EAA

###1. BayEnv


###2. LFMM



##Comparison between the different datasets & different analyses (Venn Diagram)





##Genomic Turnover: Gradient Forest






##Genomic Turnover: GDM






##Other Analyses??
