#SE: Final dataset

After speaking with Victoria, I will include all the populations (even with <5 individuals per population) in the final dataset. Also, based on population structure analyses, I will exclude the DE populations, as they are from a different gene pool. 

##Step 1: Remove DE

```
vcftools --vcf SEFinalc94d6m4p3.vcf --recode --recode-INFO-all --out SEFinalc94d6m4p3.vcf
bcftools query -l SEFinalc94d6m4p3.vcf

bcftools reheader SEFinalc94d6m4p3.vcf.recode.vcf -s raw.SE193.names --out SE193.raw.vcf

vcftools --vcf SE193.raw.vcf --remove DE.names --recode --recode-INFO-all --out SE163
```

##Step 2: Remove loci genotyped in <80% individuals

```
vcftools --vcf SE163.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out SE163.s1

After filtering, kept 16640 out of a possible 1300946 Sites
```

Look at missingness
```
vcftools --vcf SE163.s1.recode.vcf --missing-indv

##R
library(ggplot2)
SE.163.s1 <- read.table("out.imiss", header=T)
pop <- read.table("SE163.popnames", header=T)
SE.163.s1$pop <- pop$pop
SE.163.s1$pop.order <- pop$pop.order

SE.163.s1.sort <- SE.163.s1[order(SE.163.s1$pop.order),]

SE.163.s1.sort$pop <- factor(SE.163.s1.sort$pop, levels=SE.163.s1.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.163.s1.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][s1.missing]
[s1.missing]:https://cloud.githubusercontent.com/assets/12142475/20754370/dd1ec17a-b70a-11e6-977d-08cf89dd767e.png


Calculate SFS & LD for s1
```
vcftools --vcf SE163.s1.recode.vcf --plink --out SE163.s1.plink 
plink --file SE163.s1.plink --recode --recodeA --out SE163.plink

plink --file SE163.s1.plink --freq --out SE163.s1.plink
plink --file SE163.s1.plink --r2 --out SE163.s1.plink
```


##Step 3-5 MAF, HWE, LD, thin

After each step, calculate LD & SFS
```
vcftools --vcf SE163.s1.recode.vcf --maf 0.05 --recode --recode-INFO-all --out SE163.s2

After filtering, kept 3948 out of a possible 16640 Sites

vcftools --vcf SE163.s2.recode.vcf --plink --out SE163.s2.plink 
plink --file SE163.s2.plink --recode --recodeA --out SE163.s2.plink

plink --file SE163.s2.plink --freq --out SE163.s2.plink
plink --file SE163.s2.plink --r2 --out SE163.s2.plink
```

Calculate HWE and LD
```
mkdir popnames.plink.folder ##directory with 15 files each containing the population names
mkdir subset.data
for i in $(ls popnames.plink.folder/); do plink --file SE163.s2.plink --keep popnames.plink.folder/$i --recode --recodeA --out subset.data/$i.plink; done

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --freq --out subset.data/$i; done
for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --r2 --out subset.data/$i; done

##R
Sk.Ho.frq <- read.table("subset.data/Sk.Ho.frq", header=T)
Sk.SF.frq <- read.table("subset.data/Sk.SF.frq", header=T)
Sk.SL.frq <- read.table("subset.data/Sk.SL.frq", header=T)

Upp.Gra.frq <- read.table("subset.data/Upp.Gra.frq", header=T)
Upp.K.frq <- read.table("subset.data/Upp.K.frq", header=T)
Upp.O.frq <- read.table("subset.data/Upp.O.frq", header=T)

Um.Taf.frq <- read.table("subset.data/Um.Taf.frq", header=T)
Um.Gr.frq <- read.table("subset.data/Um.Gr.frq", header=T)
Um.UM3.frq <- read.table("subset.data/Um.UM3.frq", header=T)

LT1.frq <- read.table("subset.data/LT1.frq", header=T)
LT2.frq <- read.table("subset.data/LT2.frq", header=T)
LT3.frq <- read.table("subset.data/LT3.frq", header=T)

Kir.L.frq <- read.table("subset.data/Kir.L.frq", header=T)
Kir.G.frq <- read.table("subset.data/Kir.G.frq", header=T)

FIN.frq <- read.table("subset.data/FIN.frq", header=T)

par(mfrow=c(5,3))
hist(Sk.Ho.frq$MAF, main="Sk.Ho (20, 3918, 81.6% genotyping) SFS")
hist(Sk.SF.frq$MAF, main="Sk.SF (10, 3918, 95.8%) SFS")
hist(Sk.SL.frq$MAF, main="Sk.SL (19, 3918, 91.3%) SFS")

hist(Upp.Gra.frq$MAF, main="Upp.Gra (10, 3918, 86.5%) SFS")
hist(Upp.K.frq$MAF, main="Upp.K (10, 3918, 96.3%) SFS")
hist(Upp.O.frq$MAF, main="Upp.O (10, 3918, 84.9%) SFS")

hist(Um.UM3.frq$MAF, main="Umea.UT3 (4, 3918, 55%) SFS")
hist(Um.Taf.frq$MAF, main="Umea.Taf (10, 3918, 92.3%) SFS")
hist(Um.Gr.frq$MAF, main="Umea.Gr (10, 3918, 90.9%) SFS")

hist(LT1.frq$MAF, main="LT1 (10, 3918, 92.5%) SFS")
hist(LT2.frq$MAF, main="LT2 (10, 3918, 58.2%) SFS")
hist(LT3.frq$MAF, main="LT3 (10, 3918, 80.4%) SFS")

hist(Kir.G.frq$MAF, main="Kir.G (10, 3918, 96.3%) SFS")
hist(Kir.L.frq$MAF, main="Kir.L (10, 3918, 94.7%) SFS")

hist(FIN.frq$MAF, main="FIN (10, 3918, 75.4%) SFS")
```

![alt_txt][SFS.s2]
[SFS.s2]:https://cloud.githubusercontent.com/assets/12142475/20756148/dd752716-b711-11e6-84bb-20cad8635e76.png

Only variable loci
```
#Create a list of all the .frq dataframes
SE.s2.freq.list <- setNames(lapply(ls(pattern=".frq"), function(x) get(x)), ls(pattern=".frq"))

#and remove all .frq from the global env
rm(list=ls(pattern=".frq"))

#Filter all for MAF>0.001
SE.s2.freq.var.list <- lapply(SE.s2.freq.list, subset, MAF>0.01)

#plot hist of all new MAF with title
par(mfrow=c(3,5))
for(i in names(SE.s2.freq.var.list))
{
    df1 = as.data.frame(SE.s2.freq.var.list[[i]])
    hist(df1$MAF, main=i)
}

```

![alt_txt][SFS.s2]
[SFS.s2]:https://cloud.githubusercontent.com/assets/12142475/20760255/c883f4aa-b71f-11e6-97c9-0aea039f5bf6.png

Variable loci across all populations
```
attach(SE.s2.freq.var.list)
SE.s2.var.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.var.list), get))
summary(SE.s2.var.loci.freq)

SE.s2.var.loci.freq.keep <- data.frame(table(SE.s2.var.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.var.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of variable loci across 15 pops", breaks=seq(0.5,15.5, by=1.0))
detach(SE.s2.freq.var.list)
```

![alt_txt][variable.s2]
[variable.s2]:https://cloud.githubusercontent.com/assets/12142475/20762249/09d0d094-b726-11e6-8946-1814e6810cad.png

Fixed loci across all populations
```
#Filter all for fixed loci
SE.s2.freq.fixed.list <- lapply(SE.s2.freq.list, subset, MAF<0.000001)

attach(SE.s2.freq.fixed.list)
SE.s2.fixed.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.fixed.list), get))
summary(SE.s2.fixed.loci.freq)

SE.s2.fixed.loci.freq.keep <- data.frame(table(SE.s2.fixed.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.fixed.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of fixed loci across 15 pops", breaks=seq(0,14, by=1.0))
detach(SE.s2.freq.fixed.list)
```
![alt_txt][fixed.s2]
[fixed.s2]:https://cloud.githubusercontent.com/assets/12142475/20762566/1249f51a-b727-11e6-909f-99a857a6a39d.png





##step 6: Final checks: HWE, LD, missingness

LD
```
#R write tables of fixed loci
##Actually I can just filter for MAF < 0.005, so I've changed this code
for(i in names(SE.s2.freq.fixed.list)){
write.table(SE.s2.freq.fixed.list[[i]][2], paste(i,".fixedloci", sep=""), row.names=F, col.names=F, quote=F)
}
##linux
#for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --exclude $i.frq.fixedloci --recode --recodeA --out ld.subset.data/$i.var.plink; done



##new code
##LINUX
mkdir ld.subset.data

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --maf 0.005 --recode --recodeA --out ld.subset.data/$i.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --r2 --recode --recodeA --out ld.subset.data/$i.var; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
ld.subset.path <- './ld.subset.data/'
ld.var.files <- list.files(ld.subset.path, pattern='ld$')
ld.var.files.list <- lapply(ld.var.files, read.table, header=T)
names(ld.var.files.list) <- ld.var.files
names(ld.var.files.list)

##plot ld for each population
for(i in names(ld.var.files.list))
{
    df1 = as.data.frame(ld.var.files.list[[i]])
    hist(df1$R2, main=i)
}

#Calculate frequency lf LD
SE.s3.ld.R2.08.list <- lapply(ld.var.files.list, subset, R2>0.8)
attach(SE.s3.ld.R2.08.list)
SE.s3.ld.R2.08.freq <- do.call(rbind, lapply(names(SE.s3.ld.R2.08.list), get))
summary(SE.s3.ld.R2.08.freq)

SE.s3.ld.R2.08.freq.keep <- data.frame(table(SE.s3.ld.R2.08.freq$SNP_A, SE.s3.ld.R2.08.freq$SNP_B)) 
summary(SE.s3.ld.R2.08.freq.keep)
hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))
#SE.s3.ld.R2.08.freq.keep <- subset(SE.s3.ld.R2.08.freq.keep, Freq>0) 
#hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))

```

![alt_txt][LD.all]
[LD.all]:https://cloud.githubusercontent.com/assets/12142475/20789034/49ad8328-b7b3-11e6-8e6c-e49de3f02883.png

![alt_txt][LD.freq]
[LD.freq]:https://cloud.githubusercontent.com/assets/12142475/20790452/a8f614da-b7b8-11e6-8bf4-3e414fce13d9.png


So LD between loci are mostly only in 1 population. The max is 6 (1 locus), and then 4 (6 loci). So I will not filter any of these loci out. 


HWE
Make sure that all the missing data are also filtered out of the datasets. Using the old script above, only the fixed loci per population are removed, but the ungenotyped loci remain. So there will be gaps in the .hwe file, and this can't be imported into R. 
```
##Linux
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --maf 0.005 --out ld.subset.data/$ivar.plink; done
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --hardy --out ld.subset.data/$i; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
hwe.subset.path <- './ld.subset.data/'
hwe.files <- list.files(hwe.subset.path, pattern='hwe$')
hwe.files
setwd('./ld.subset.data')
hwe.files.list <- lapply(hwe.files, read.table, header=T)
names(hwe.files.list) <- hwe.files  ##name each dataframe

#subset the data to include only TEST=="ALL"
hwe.files.list <- lapply(hwe.files.list, subset, TEST=="ALL")

#create a list of df with all loci out of HWE and O.Het>0.5
hwe.files.list.remove <- lapply(hwe.files.list, subset, (P<0.050001|O.HET.>0.5))

attach(hwe.files.list.remove)
hwe.remove.freq <- do.call(rbind, lapply(names(hwe.files.list.remove), get))
summary(hwe.remove.freq)
detach(hwe.files.list.remove)

hwe.remove.freq.freq <- data.frame(table(hwe.remove.freq$SNP)) 
summary(hwe.remove.freq.freq)
hist(hwe.remove.freq.freq$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of loci deviating from HWE and O.Het>0.5 across 15 pops"))

```

![alt_txt][HWE.freq]
[HWE.freq]:https://cloud.githubusercontent.com/assets/12142475/20799028/32c4e7fe-b7e1-11e6-868c-4b7d2fca003a.png


137 Loci to remove
```
#R
HWE.loc.remove <- subset(hwe.remove.freq.freq, Freq>5)
summary(HWE.loc.remove)

setwd("/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter")
write.table(HWE.loc.remove$Var1, "HWE.loc.remove", col.name=F, row.name=F, quote=F)


#Linux
plink --file SE163.s3.plink --exclude HWE.loc.remove --recode --recodeA --out SE163.s4.plink
```
Convert to vcf using pgdspider





Missingness

```
vcftools --vcf SE163.s4.vcf --recode --recode-INFO-all --out SE163.s4
vcftools --vcf SE163.s4.recode.vcf --missing-indv

##R
library(ggplot2)
SE.163.s4 <- read.table("out.imiss", header=T)
pop <- read.table("SE163.popnames", header=T)
SE.163.s4$pop <- pop$pop
SE.163.s4$pop.order <- pop$pop.order

SE.163.s4.sort <- SE.163.s4[order(SE.163.s4$pop.order),]

SE.163.s4.sort$pop <- factor(SE.163.s4.sort$pop, levels=SE.163.s4.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.163.s4.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][missing.s4]
[missing.s4]:https://cloud.githubusercontent.com/assets/12142475/20800272/3763c2ea-b7e5-11e6-9cba-30385821345c.png


remove individuals with >40% missingness
```
SE163.imiss.40 <- subset(SE.163.s4, F_MISS>0.4)
summary(SE163.imiss.40)
write.table(SE163.imiss.40$INDV, "SE163.imiss40", col.name=F, row.name=F, quote=F)

#LINUX
vcftools --vcf SE163.s4.recode.vcf --remove SE163.imiss40 --recode --recode-INFO-all --out SE145.FINAL

##rename individuals
bcftools reheader SE145.FINAL.recode.vcf -s names.SE145FINAL --out SE145.FINAL.vcf

#convert to plink
vcftools --vcf SE145.FINAL.vcf --plink --out SE145.FINAL.plink
plink --file SE145.FINAL.plink --recode --recodeA --out SE145.FINAL.plink

##subset into populations
mkdir FINAL.subset.data

for i in $(ls popnames.plink.folder/); do plink --file SE145.FINAL.plink --keep popnames.plink.folder/$i --recode --recodeA --out FINAL.subset.data/$i.FINAL.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.plink --maf 0.005 --recode --recodeA --out FINAL.subset.data/$i.FINAL.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --r2 --out FINAL.subset.data/$i.FINAL.var; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --freq --out FINAL.subset.data/$i.FINAL.var; done


##Final graphs of per population SFS and LD
##LD
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.ld.files <- list.files(FINAL.subset.path, pattern='ld$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/FINAL.subset.data/')
FINAL.ld.files.list <- lapply(FINAL.ld.files, read.table, header=T)
names(FINAL.ld.files.list) <- FINAL.ld.files
names(FINAL.ld.files.list)

##plot ld for each population
par(mfrow=c(5,3))
for(i in names(FINAL.ld.files.list))
{
    df1 = as.data.frame(FINAL.ld.files.list[[i]])
    hist(df1$R2, main=i)
}

##SFS
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.sfs.files <- list.files(FINAL.subset.path, pattern='var.frq$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/FINAL.subset.data/')
FINAL.sfs.files.list <- lapply(FINAL.sfs.files, read.table, header=T)
names(FINAL.sfs.files.list) <- FINAL.sfs.files
names(FINAL.sfs.files.list)

##plot ld for each population
par(mfrow=c(5,3))
for(i in names(FINAL.sfs.files.list))
{
    df1 = as.data.frame(FINAL.sfs.files.list[[i]])
    hist(df1$MAF, main=i)
}


```

![alt_txt][FINAL.ld.perpop]
[FINAL.ld.perpop]:https://cloud.githubusercontent.com/assets/12142475/20802839/9c6f5f34-b7ed-11e6-8c88-5ad301fcfb40.png

![alt_txt][FINAL.sfs.perpop]
[FINAL.sfs.perpop]:https://cloud.githubusercontent.com/assets/12142475/20803087/522c3950-b7ee-11e6-8e07-b27ce35959ae.png


