#SE: Final dataset

After speaking with Victoria, I will include all the populations (even with <5 individuals per population) in the final dataset. Also, based on population structure analyses, I will exclude the DE populations, as they are from a different gene pool. 

I'm also removing 12 Skane individuals that look different from the rest. I think something went wrong when I concatenated data from different runs. These individuals all tend to have higher genetic diversity than expected. I will have to map these data to determine what the correct SNP calls are. 

Diversity and F-statistics will be based on the whole dataset for now. I will reassess after speaking with Victoria. 

##Step 1: Remove DE

And Sk individuals. 

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133

```
vcftools --vcf SEFinalc94d6m4p3.vcf --recode --recode-INFO-all --out SEFinalc94d6m4p3.vcf
bcftools query -l SEFinalc94d6m4p3.vcf

bcftools reheader SEFinalc94d6m4p3.vcf.recode.vcf -s raw.SE193.names --out SE193.raw.vcf

vcftools --vcf SE193.raw.vcf --remove DE.Sk.names --recode --recode-INFO-all --out SE151
```

##Step 2: Remove loci genotyped in <80% individuals

```
vcftools --vcf SE151.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out SE151.s1

After filtering, kept 18800 out of a possible 1300946 Sites
```

Look at missingness
```
vcftools --vcf SE151.s1.recode.vcf --missing-indv

##R
library(ggplot2)
SE.151.s1 <- read.table("out.imiss", header=T)
pop <- read.table("SE151.popnames", header=T)
SE.151.s1$pop <- pop$pop
SE.151.s1$pop.order <- pop$pop.order

SE.151.s1.sort <- SE.151.s1[order(SE.151.s1$pop.order),]

SE.151.s1.sort$pop <- factor(SE.151.s1.sort$pop, levels=SE.151.s1.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.151.s1.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][s1.missing]
[s1.missing]:https://cloud.githubusercontent.com/assets/12142475/20881932/eff4ffee-bade-11e6-80a6-5d3ff2f83310.png


Calculate SFS & LD for s1
```
vcftools --vcf SE151.s1.recode.vcf --plink --out SE151.s1.plink 
plink --file SE151.s1.plink --recode --recodeA --out SE151.plink

plink --file SE151.s1.plink --freq --out SE151.s1.plink
plink --file SE151.s1.plink --r2 --out SE151.s1.plink
```


##Step 3-5 MAF, HWE, LD, thin

After each step, calculate LD & SFS
```
vcftools --vcf SE151.s1.recode.vcf --maf 0.05 --recode --recode-INFO-all --out SE151.s2

After filtering, kept 3897 out of a possible 18800 Sites

vcftools --vcf SE151.s2.recode.vcf --plink --out SE151.s2.plink 
plink --file SE151.s2.plink --recode --recodeA --out SE151.s2.plink

plink --file SE151.s2.plink --freq --out SE151.s2.plink
plink --file SE151.s2.plink --r2 --out SE151.s2.plink
```

Calculate HWE and LD
```
mkdir popnames.plink.folder ##directory with 15 files each containing the population names
mkdir subset.data
for i in $(ls popnames.plink.folder/); do plink --file SE151.s2.plink --keep popnames.plink.folder/$i --recode --recodeA --out subset.data/$i.plink; done

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --freq --out subset.data/$i; done
for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --r2 --out subset.data/$i; done

##R
Sk.Ho.frq <- read.table("subset.data/Sk.Ho.frq", header=T)
Sk.SF.frq <- read.table("subset.data/Sk.SF.frq", header=T)
Sk.SL.frq <- read.table("subset.data/Sk.SL.frq", header=T)

Upp.Gra.frq <- read.table("subset.data/Upp.Gra.frq", header=T)
Upp.K.frq <- read.table("subset.data/Upp.K.frq", header=T)
Upp.O.frq <- read.table("subset.data/Upp.O.frq", header=T)

Um.Taf.frq <- read.table("subset.data/Um.Taf.frq", header=T)
Um.Gr.frq <- read.table("subset.data/Um.Gr.frq", header=T)
Um.UM3.frq <- read.table("subset.data/Um.UM3.frq", header=T)

LT1.frq <- read.table("subset.data/LT1.frq", header=T)
LT2.frq <- read.table("subset.data/LT2.frq", header=T)
LT3.frq <- read.table("subset.data/LT3.frq", header=T)

Kir.L.frq <- read.table("subset.data/Kir.L.frq", header=T)
Kir.G.frq <- read.table("subset.data/Kir.G.frq", header=T)

FIN.frq <- read.table("subset.data/FIN.frq", header=T)

par(mfrow=c(3.5))
hist(Sk.Ho.frq$MAF, main="Sk.Ho (10, 3874, 85.3% genotyping) SFS")
hist(Sk.SF.frq$MAF, main="Sk.SF (10, 3874, 95.5%) SFS")
hist(Sk.SL.frq$MAF, main="Sk.SL (17, 3874, 92.6%) SFS")

hist(Upp.Gra.frq$MAF, main="Upp.Gra (10, 3874, 85.9%) SFS")
hist(Upp.K.frq$MAF, main="Upp.K (10, 3874, 95.8%) SFS")
hist(Upp.O.frq$MAF, main="Upp.O (10, 3874, 84.3%) SFS")

hist(Um.UM3.frq$MAF, main="Umea.UT3 (4, 3874, 53.3%) SFS")
hist(Um.Taf.frq$MAF, main="Umea.Taf (10, 3874, 91.8%) SFS")
hist(Um.Gr.frq$MAF, main="Umea.Gr (10, 3874, 90.6%) SFS")

hist(LT1.frq$MAF, main="LT1 (10, 3874, 91.8%) SFS")
hist(LT2.frq$MAF, main="LT2 (10, 3874, 57.2%) SFS")
hist(LT3.frq$MAF, main="LT3 (10, 3874, 79.3%) SFS")

hist(Kir.G.frq$MAF, main="Kir.G (10, 3874, 95.9%) SFS")
hist(Kir.L.frq$MAF, main="Kir.L (10, 3874, 94.2%) SFS")

hist(FIN.frq$MAF, main="FIN (10, 3874, 74.8%) SFS")
```


![alt_txt][SFS.s2.new]
[SFS.s2.new]:https://cloud.githubusercontent.com/assets/12142475/20882278/721773b6-bae0-11e6-8fb4-0c21de3acc85.png



Only variable loci
```
#Create a list of all the .frq dataframes
SE.s2.freq.list <- setNames(lapply(ls(pattern=".frq"), function(x) get(x)), ls(pattern=".frq"))

#and remove all .frq from the global env
rm(list=ls(pattern=".frq"))

#Filter all for MAF>0.001
SE.s2.freq.var.list <- lapply(SE.s2.freq.list, subset, MAF>0.01)

#plot hist of all new MAF with title
par(mfrow=c(3,5))
for(i in names(SE.s2.freq.var.list))
{
    df1 = as.data.frame(SE.s2.freq.var.list[[i]])
    hist(df1$MAF, main=i)
}

```

![alt_txt][SFS.s2]
[SFS.s2]:https://cloud.githubusercontent.com/assets/12142475/20882391/04a5595a-bae1-11e6-87d5-a6665866e982.png

Variable loci across all populations
```
attach(SE.s2.freq.var.list)
SE.s2.var.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.var.list), get))
summary(SE.s2.var.loci.freq)

SE.s2.var.loci.freq.keep <- data.frame(table(SE.s2.var.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.var.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of variable loci across 15 pops", breaks=seq(0.5,15.5, by=1.0))
detach(SE.s2.freq.var.list)
```

![alt_txt][variable.s2]
[variable.s2]:https://cloud.githubusercontent.com/assets/12142475/20882428/30704716-bae1-11e6-8f18-b331bfbb02aa.png

Fixed loci across all populations
```
#Filter all for fixed loci
SE.s2.freq.fixed.list <- lapply(SE.s2.freq.list, subset, MAF<0.000001)

attach(SE.s2.freq.fixed.list)
SE.s2.fixed.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.fixed.list), get))
summary(SE.s2.fixed.loci.freq)

SE.s2.fixed.loci.freq.keep <- data.frame(table(SE.s2.fixed.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.fixed.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of fixed loci across 15 pops", breaks=seq(0,14, by=1.0))
detach(SE.s2.freq.fixed.list)
```
![alt_txt][fixed.s2]
[fixed.s2]:https://cloud.githubusercontent.com/assets/12142475/20882453/4acd1878-bae1-11e6-97d3-b227c39b4c6d.png

##Thin to 1 SNP per locus
```
vcftools --vcf SE151.s2.recode.vcf --thin 130 --recode --recode-INFO-all --out SE151.s3

After filtering, kept 2192 out of a possible 3897 Sites

vcftools --vcf SE151.s3.recode.vcf --plink --out SE151.s3.plink
```



##step 6: Final checks: HWE, LD, missingness

LD
```
#R write tables of fixed loci
##Actually I can just filter for MAF < 0.005, so I've changed this code
for(i in names(SE.s3.freq.fixed.list)){
write.table(SE.s3.freq.fixed.list[[i]][2], paste(i,".fixedloci", sep=""), row.names=F, col.names=F, quote=F)
}
##linux
#for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --exclude $i.frq.fixedloci --recode --recodeA --out ld.subset.data/$i.var.plink; done



##new code
##LINUX
mkdir ld.subset.data

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --maf 0.005 --recode --recodeA --out ld.subset.data/$i.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --r2 --out ld.subset.data/$i.var; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/')
ld.subset.path <- './ld.subset.data/'
ld.var.files <- list.files(ld.subset.path, pattern='ld$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/ld.subset.data/')
ld.var.files.list <- lapply(ld.var.files, read.table, header=T)
names(ld.var.files.list) <- ld.var.files
names(ld.var.files.list)

##plot ld for each population
par(mfrow=c(3,5))
for(i in names(ld.var.files.list))
{
    df1 = as.data.frame(ld.var.files.list[[i]])
    hist(df1$R2, main=i)
}

#Calculate frequency lf LD
SE.s3.ld.R2.08.list <- lapply(ld.var.files.list, subset, R2>0.8)
attach(SE.s3.ld.R2.08.list)
SE.s3.ld.R2.08.freq <- do.call(rbind, lapply(names(SE.s3.ld.R2.08.list), get))
summary(SE.s3.ld.R2.08.freq)

SE.s3.ld.R2.08.freq.keep <- data.frame(table(SE.s3.ld.R2.08.freq$SNP_A, SE.s3.ld.R2.08.freq$SNP_B)) 
summary(SE.s3.ld.R2.08.freq.keep)
hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))
#SE.s3.ld.R2.08.freq.keep <- subset(SE.s3.ld.R2.08.freq.keep, Freq>1) 
#hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))

```

![alt_txt][LD.all]
[LD.all]:https://cloud.githubusercontent.com/assets/12142475/20882834/2bbd5d92-bae3-11e6-989c-74ab9d283f68.png

Most of the loci in LD are found only in 1 population. 

![alt_txt][LD.freq]
[LD.freq]:https://cloud.githubusercontent.com/assets/12142475/20882899/89ca3c0c-bae3-11e6-8c5d-bc0a23f47ca5.png

In >1pop: 

![alt_txt][LD.freq.HighFreq]
[LD.freq.HighFreq]:https://cloud.githubusercontent.com/assets/12142475/20882937/c19ea5dc-bae3-11e6-868d-fb6e65d46097.png



So LD between loci are mostly only in 1 population. The max is 5 (1 locus). So I will not filter any of these loci out. 


HWE
Make sure that all the missing data are also filtered out of the datasets. Using the old script above, only the fixed loci per population are removed, but the ungenotyped loci remain. So there will be gaps in the .hwe file, and this can't be imported into R. 
```
##Linux
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --maf 0.005 --out ld.subset.data/$ivar.plink; done
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --hardy --out ld.subset.data/$i; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/')
hwe.subset.path <- './ld.subset.data/'
hwe.files <- list.files(hwe.subset.path, pattern='hwe$')
hwe.files
setwd('./ld.subset.data')
hwe.files.list <- lapply(hwe.files, read.table, header=T)
names(hwe.files.list) <- hwe.files  ##name each dataframe

#subset the data to include only TEST=="ALL"
hwe.files.list <- lapply(hwe.files.list, subset, TEST=="ALL")

#create a list of df with all loci out of HWE and O.Het>0.5
hwe.files.list.remove <- lapply(hwe.files.list, subset, (P<0.050001|O.HET.>0.5))

attach(hwe.files.list.remove)
hwe.remove.freq <- do.call(rbind, lapply(names(hwe.files.list.remove), get))
summary(hwe.remove.freq)
detach(hwe.files.list.remove)

hwe.remove.freq.freq <- data.frame(table(hwe.remove.freq$SNP)) 
summary(hwe.remove.freq.freq)
hist(hwe.remove.freq.freq$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of loci deviating from HWE and O.Het>0.5 across 15 pops")

```

![alt_txt][HWE.freq]
[HWE.freq]:https://cloud.githubusercontent.com/assets/12142475/20883013/26f22882-bae4-11e6-8232-8189b4ed0e08.png


153 Loci to remove
```
#R
HWE.loc.remove <- subset(hwe.remove.freq.freq, Freq>5)
summary(HWE.loc.remove)

setwd("/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/")
write.table(HWE.loc.remove$Var1, "HWE.loc.remove", col.name=F, row.name=F, quote=F)


#Linux
plink --file SE151.s3.plink --exclude HWE.loc.remove --recode --recodeA --out SE151.s4.plink

#2081SNPs
```


Convert to vcf using pgdspider






Missingness
```
vcftools --vcf SE151.s4.vcf --recode --recode-INFO-all --out SE151.s4
vcftools --vcf SE151.s4.recode.vcf --missing-indv

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/')
library(ggplot2)
SE.151.s4 <- read.table("out.imiss", header=T)
pop <- read.table("SE151.popnames", header=T)
SE.151.s4$pop <- pop$pop
SE.151.s4$pop.order <- pop$pop.order

SE.151.s4.sort <- SE.151.s4[order(SE.151.s4$pop.order),]

SE.151.s4.sort$pop <- factor(SE.151.s4.sort$pop, levels=SE.151.s4.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.151.s4.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][missing.s4]
[missing.s4]:https://cloud.githubusercontent.com/assets/12142475/20883961/0ad3a2f2-bae9-11e6-8a81-eabc03bbc9fe.png


remove individuals with >40% missingness
```
SE151.imiss.40 <- subset(SE.151.s4, F_MISS>0.4)
summary(SE151.imiss.40)
write.table(SE151.imiss.40$INDV, "SE151.imiss40", col.name=F, row.name=F, quote=F)

#LINUX
vcftools --vcf SE151.s4.recode.vcf --remove SE151.imiss40 --recode --recode-INFO-all --out SE132.FINAL

##rename individuals
bcftools reheader SE132.FINAL.recode.vcf -s names.SE132.FINAL --out SE132.FINAL.vcf
```




```
#convert to plink
vcftools --vcf SE132.FINAL.vcf --plink --out SE132.FINAL.plink
plink --file SE132.FINAL.plink --recode --recodeA --out SE132.FINAL.plink

##subset into populations
mkdir FINAL.subset.data

for i in $(ls popnames.plink.folder/); do plink --file SE132.FINAL.plink --keep popnames.plink.folder/$i --recode --recodeA --out FINAL.subset.data/$i.FINAL.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.plink --maf 0.05 --recode --recodeA --out FINAL.subset.data/$i.FINAL.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --r2 --out FINAL.subset.data/$i.FINAL.var; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --freq --out FINAL.subset.data/$i.FINAL.var; done


##R: Final graphs of per population SFS and LD
##LD
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.ld.files <- list.files(FINAL.subset.path, pattern='ld$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/FINAL.subset.data/')
FINAL.ld.files.list <- lapply(FINAL.ld.files, read.table, header=T)
names(FINAL.ld.files.list) <- FINAL.ld.files
names(FINAL.ld.files.list)

##plot ld for each population
par(mfrow=c(3,5))
for(i in names(FINAL.ld.files.list))
{
    df1 = as.data.frame(FINAL.ld.files.list[[i]])
    hist(df1$R2, main=i)
}

##SFS
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.sfs.files <- list.files(FINAL.subset.path, pattern='var.frq$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE133/FINAL.subset.data/')
FINAL.sfs.files.list <- lapply(FINAL.sfs.files, read.table, header=T)
names(FINAL.sfs.files.list) <- FINAL.sfs.files
names(FINAL.sfs.files.list)

##plot ld for each population
par(mfrow=c(3,5))
for(i in names(FINAL.sfs.files.list))
{
    df1 = as.data.frame(FINAL.sfs.files.list[[i]])
    hist(df1$MAF, main=i)
}


```

![alt_txt][FINAL.ld.perpop]
[FINAL.ld.perpop]:https://cloud.githubusercontent.com/assets/12142475/20884085/a7e47616-bae9-11e6-8fa2-101aedd4ab8d.png

![alt_txt][FINAL.sfs.perpop]
[FINAL.sfs.perpop]:https://cloud.githubusercontent.com/assets/12142475/20884119/c8e27f5c-bae9-11e6-92e9-cc4e460af2e6.png


####FINAL DATA

SE132.FINAL.plink /.vcf

132 individuals

2081 loci

15 populations

6 regions


##Population Structure

###Fst

Convert plink FINAL to structure using pgdspider, and import into R
```
/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/SumStats

library(adegenet)
library(hierfstat)
library(reshape)

SEall.132 <- read.structure("SE132.FINAL.str")
SEall.132

pop.SE132 <- read.table("popnames.SE132.structure", header=T)  ##make sure the populations are numbered "X1.Sk.Ho", etc.
pop.factor <- as.factor(pop.SE132$pop)
SEall.132@pop <- pop.factor

hier.SEall <- genind2hierfstat(SEall.132)

SEall.fst <- pairwise.fst(SEall.132, pop=NULL, res.type=c("dist", "matrix"))

m <- as.matrix(SEall.fst)
m2 <- melt(m)[melt(upper.tri(m))$value,]
names(m2)<- c("c1","c2", "distance")

library(gplots)

shadesOfGrey <- colorRampPalette(c("grey100", "grey0"))  ##define the colourpalette. 

Dend <- read.table("heatmap.popcolours", header=T)  ##list of colour names for each population based on R colour palatte. In alphabetical order (as in genind file)
Dend.Colours <- as.character(Dend$colours.pop)

par(oma=c(1,1,2,1))
heatmap.2(as.matrix(SEall.fst), trace="none", RowSideColors=Dend.Colours, ColSideColors=Dend.Colours, col=shadesOfGrey, labRow=F, labCol=F, key.ylab=NA, key.xlab=NA, key.title="Fst Colour Key", keysize=0.9, main="Pairwise Fst of SEall: 15pops, 6regions, 2081loci")  ##RowSideColors is for the dendrogram on the row, ColSideColors for the upper dendrogram. Colour order should be the same as the input. The pop order is alphabetical in the output. 
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")

popnames.all <- as.character(Dend$pop)
legend("bottom", popnames.all, xpd = TRUE, horiz = TRUE, inset = c(0, 0), bty="o", pch=15, col=Dend.Colours, title="Side Dendrogram:Region")
```

![alt_txt][Fst.SEall]
[Fst.SEall]:https://cloud.githubusercontent.com/assets/12142475/20884888/26b43e00-baee-11e6-9f84-3067feac8af8.png

####Frequency distributions of Fst per locus

This is to find loci that explain differentiation between populations more than neutral expectation


```
#Convert plink .ped files to structure using pgdspider
#create population files 

#R
library(adegenet)
library(hierfstat)
SE132.genind <- read.structure("SE132.plink.str")
hier.SE132 <- genind2hierfstat(SE132.genind)
SE132.genind@pop
SE132.pop <- read.table("SE132.pop", header=T)
SE132.pop.factor <- as.factor(SE132.pop$pop)
SE132.genind@pop <- SE132.pop.factor
SE132.genind@pop
hier.SE132 <- genind2hierfstat(SE132.genind)
stats.SE132 <- basic.stats(hier.SE132)
stats.SE132.perlocus <- stats.SE132$perloc

hist(stats.SE132.perlocus$Fst, xlim=c(-0.2, 1.0), breaks=120)
```

![alt_txt][Fst.perloc]
[Fst.perloc]:https://cloud.githubusercontent.com/assets/12142475/20885005/ce2dd128-baee-11e6-83be-a043cb171c9c.png


Write table with names of loci with Fst>0.5
```
SE132.Fstouliers <- subset(stats.SE132.perlocus, Fst>0.5)
summary(SE132.Fstouliers)

write.table(SE132.Fstouliers, "SE132.Fstoutliers", quote=F, sep=" ")
```

63 outliers in total from the per locus Fst

###IBD

Fst/(1-Fst) vs log.dist(km) - according to Rousset et al. 1997, this is correlated with the effective population density (De) and effective dispersal distance (variance)

```
##Fst 15 pops -> Fst/(1-Fst)

library(reshape)
library(fields)


m <- as.matrix(SEall.fst)
m
m2 <- melt(m)[melt(upper.tri(m))$value,]
names(m2) <- c("c1", "c2", "distance")
m2
m2$IBD <- m2$distance/(1-m2$distance)


SE.pop.coords <- read.table("SE.coords.15pop", header=T)
SEpop_lon.lat <- cbind(SE.pop.coords$Long, SE.pop.coords$Lat)
distance.matrix.SEpop <- rdist.earth(SEpop_lon.lat, miles=F)  ##great circle dist based on the coordinates
m.dist <- as.matrix(distance.matrix.SEpop)
summary(m.dist)

m2.dist <- melt(m.dist)[melt(upper.tri(m.dist))$value,]
names(m2.dist) <- c("c1", "c2", "distance")
summary(m2.dist)
m2.dist$log.km <- log(m2.dist$distance)


library(MASS)
#dens <- kde2d(m2$IBD,m2.dist$log.km, n=10)
#myPal <- colorRampPalette(c("white","blue","gold", "orange", "red"))
plot(m2$IBD~m2.dist$log.km, pch=20,cex=.5, xlab="log Geographic distance (km)", ylab="Fst/(1-Fst)")
#image(dens, col=transp(myPal(10),.7), add=TRUE)
abline(fit <- lm(m2$IBD~m2.dist$log.km))
legend("bottomright", bty="n", legend=paste("R2 =", format(summary(fit)$adj.r.squared, digits=4)))  ##and paste R2
title("Isolation by distance plot - SEall")
```

![alt_txt][IBD.SEall]
[IBD.SEall]:https://cloud.githubusercontent.com/assets/12142475/20885224/f1579fc0-baef-11e6-9f99-fcd105098e03.png



###AMOVA

http://grunwaldlab.github.io/Population_Genetics_in_R/AMOVA.html

```
library(adegenet)
library(poppr)

SE132.strata <- pop.SE132[,1:3]  ##from text file. each column has one hierarchy level specified for all individuals. (indiv, pop, region)
SEall.132.genind <- SEall.132
SEall.132.genind@other <- SE132.strata ##renamed the SEall.132 genind to a better name

strata(SEall.132.genind) <- other(SEall.132.genind)

SEall.132.genclone <- as.genclone(SEall.132.genind)

SE132.amova <- poppr.amova(SEall.132.genclone, ~region/pop)

SE132.amova 
$call
ade4::amova(samples = xtab, distances = xdist, structures = xstruct)

$results
                            Df    Sum Sq    Mean Sq
Between region               5  5624.345 1124.86904
Between pop Within region    9  1563.566  173.72952
Between samples Within pop 117 10348.789   88.45119
Within samples             132 11068.430   83.85174
Total                      263 28605.130  108.76475

$componentsofcovariance
                                            Sigma          %
Variations  Between region              22.075461  19.481964
Variations  Between pop Within region    5.085375   4.487929
Variations  Between samples Within pop   2.299721   2.029542
Variations  Within samples              83.851744  74.000566
Total variations                       113.312301 100.000000

$statphi
                         Phi
Phi-samples-total 0.25999434
Phi-samples-pop   0.02669392
Phi-pop-region    0.05573818
Phi-region-total  0.19481964

SE132.amovatest <- randtest(SE145.amova, nrepet=999)

SE132.amovatest

class: krandtest 
Monte-Carlo tests
Call: randtest.amova(xtest = SE132.amova, nrepet = 999)

Number of tests:   4 

Adjustment method for multiple comparisons:   none 
Permutation number:   999 
                        Test       Obs    Std.Obs   Alter Pvalue
1  Variations within samples 83.851744 -15.211099    less  0.001
2 Variations between samples  2.299721   1.405146 greater  0.079
3     Variations between pop  5.085375  20.240561 greater  0.001
4  Variations between region 22.075461   6.671363    less  1.000


##Randomised samples
SE132.random <- SEall.132.genclone
set.seed(9001)
strata(SE132.random) <- strata(SEall.132.genclone)[sample(nInd(SEall.132.genclone)), -1]
head(strata(SE132.random))
head(strata(SEall.132.genclone))
SE132.random.amova <- poppr.amova(SE132.random, ~region/pop)

SE132.random.amova   ##now all the variation is within samples and within populations. So no population structure evident. 
$call
ade4::amova(samples = xtab, distances = xdist, structures = xstruct)

$results
                            Df     Sum Sq   Mean Sq
Between region               5   699.3817 139.87634
Between pop Within region    9  1263.0980 140.34422
Between samples Within pop 117 15574.2199 133.11299
Within samples             132 11068.4302  83.85174
Total                      263 28605.1298 108.76475

$componentsofcovariance
                                              Sigma            %
Variations  Between region              -0.02962594  -0.02720872
Variations  Between pop Within region    0.43121766   0.39603415
Variations  Between samples Within pop  24.63062339  22.62098440
Variations  Within samples              83.85174396  77.01019018
Total variations                       108.88395908 100.00000000

$statphi
                            Phi
Phi-samples-total  0.2298980982
Phi-samples-pop    0.2270472519
Phi-pop-region     0.0039592642
Phi-region-total  -0.0002720872


SE132.random.amovatest<- randtest(SE132.random.amova, nrepet=999) 

SE132.random.amovatest

class: krandtest 
Monte-Carlo tests
Call: randtest.amova(xtest = SE132.random.amova, nrepet = 999)

Number of tests:   4 

Adjustment method for multiple comparisons:   none 
Permutation number:   999 
                        Test         Obs     Std.Obs   Alter Pvalue
1  Variations within samples 83.85174396 -16.4837507    less  0.001
2 Variations between samples 24.63062339  14.8702325 greater  0.001
3     Variations between pop  0.43121766   0.6740932 greater  0.228
4  Variations between region -0.02962594  -0.2969393    less  0.409

other elements: adj.method call 


##plot both outputs
par(mfrow=c(2,1))
plot(SE132.amovatest)
plot(SE132.random.amovatest)
```

![alt_txt][AMOVA.SE145]
[AMOVA.SE145]:https://cloud.githubusercontent.com/assets/12142475/20885592/0b398014-baf2-11e6-83da-db379a4e48bb.png

![alt_txt][AMOVA.random]
[AMOVA.random]:https://cloud.githubusercontent.com/assets/12142475/20885605/1570c218-baf2-11e6-9a21-867c35fa379f.png


Structure between pops and between regions is higher than expected with random mating. 


###DAPC

tutorial-dapc: A tutorial for Discriminant Analysis of Principal Components (DAPC) using adegenet 2.0.0

total variance = (variance between groups) + (variance within groups)

or more simply, denoting X the data matrix:

VAR(X) = B(X) +W(X)

Usual approaches such as Principal Component Analysis (PCA) or Principal Coordinates Analysis (PCoA / MDS) focus on V AR(X). That is, they only describe the global diversity, possibly overlooking differences between groups. On the contrary, DAPC optimizes B(X) while minimizing W(X): it seeks synthetic variables, the discriminant functions, which show differences between groups as best as possible while minimizing variation within clusters.

```
##1. estimate the number of clusters

Using k-means. Which finds the number of clusters with minimises W(X) and maximises B(X). Compare using BIC

Run algorithm on PCA transformed data. I.e. reduce the dataset so that it can run faster. 

grp.SE132 <- find.clusters(SEall.132.genind, max.n.clust=40)

> choose nr of PCs: 200  ##I try to keep all the PCs

> choose k: 4 ##i used K4-6. See table below


names.15 <- c("Sk.Ho", "Sk.SF", "SK.SL", "Upp.Gra", "Upp.K", "Upp.O", "Um.Gr", "Um.Taf", "Um.UT3", "LT1", "LT2", "LT3", "Kir.G", "Kir.L", "FIN")
names.15 <- as.character(names.15)
table.value(table(pop(SEall.132.genind), grp.SE132$grp), col.lab=paste("inf", 1:4), row.lab=names.15)

dapc1.SE132 <- dapc(SEall.132.genind, grp.SE132$grp)
scatter(dapc1.SE132)
```

Nr of PCs: I kept 200

![alt_txt][PCs.initial]
[PCs.initial]:https://cloud.githubusercontent.com/assets/12142475/20885874/6f66c258-baf3-11e6-94da-a3fad9b0410a.png

Choose K: 4
![alt_txt][K.initial]
[K.initial]:https://cloud.githubusercontent.com/assets/12142475/20885882/7a3a1c0c-baf3-11e6-851c-221b9f0cc479.png


![alt_txt][K4]
[K4]:https://cloud.githubusercontent.com/assets/12142475/20885881/7a38ce56-baf3-11e6-95f1-1a33b6223d51.png

![alt_txt][K5]
[K5]:https://cloud.githubusercontent.com/assets/12142475/20885924/ba25f318-baf3-11e6-99c1-331cc8f2d088.png

![alt_txt][K6]
[K6]:https://cloud.githubusercontent.com/assets/12142475/20885923/ba255b92-baf3-11e6-96c5-14aa12ecd613.png



###PCA

PCAdapt in R:
```
##convert .vcf to plink 
##linux

vcftools --vcf SE.132.FINAL.recode.vcf --plink --out SE132.FINAL.plink

plink --file SE132.FINAL.plink --recode --recodeA --out SE132.FINAL.plink

##R
library(pcadapt)

SE.132 <- read.pcadapt("SE132.FINAL.plink.ped", type="ped")
Summary:

        - input file      SE145.FINAL.plink.ped
        - output file     SE145.FINAL.plink.pcadapt

	- number of individuals detected:	132
	- number of loci detected:		2081

File has been sucessfully converted.

##Check the nr of PCs

x <- pcadapt(SE.132, K=20)

Reading file SE132.FINAL.plink.pcadapt...
Number of SNPs: 2081
Number of individuals: 132
Number of SNPs with minor allele frequency lower than 0.05 ignored: 54
17046 out of 274692 missing data ignored.

plot(x, option="screeplot")  ##PC for pop structure = on the steep curve
```

Choose number of PCs:

![alt_txt][PCadapt.fig1]
[PCadapt.fig1]:https://cloud.githubusercontent.com/assets/12142475/20884529/1ede27c4-baec-11e6-8593-6cc980d39b90.png

K 4-5 

Plot the PCA using population information
```
pop.SE132  <- read.table("pop.SE132", header=T)
head(pop.SE132)
        indiv        pop region
1      FIN_21     X6.FIN    FIN
2      FIN_22     X6.FIN    FIN
3      FIN_23     X6.FIN    FIN
4      FIN_25     X6.FIN    FIN

poplist <- as.character(pop.SE132[,2])  ##select pops
plot(x,option="scores",pop=poplist)

poplist <- as.character(pop.SE132[,3]) ##select regions
plot(x,option="scores",pop=poplist)
```

Populations
![alt_txt][PCadapt.Fig2]
[PCadapt.Fig2]:https://cloud.githubusercontent.com/assets/12142475/20884791/ae0970d8-baed-11e6-82c3-9829e173bc27.png


Regions
![alt_txt][PCadapt.Fig3]
[PCadapt.Fig3]:https://cloud.githubusercontent.com/assets/12142475/20884801/b290de20-baed-11e6-9354-e428878eb8d7.png



##fastStructure



##TESS3


#Env Associations

##BIOclim variables

Find the uncorrelated variables from BIOclim: 

Extract BIOclim variables. See: https://github.com/alexjvr1/Manuscripts/blob/master/6.R.temp_SE.MS1.md


```
library(corrplot)
library(caret)
datMy <- read.table("SEonly.BIOclim", header=T)
datMy.scale <- scale(datMy[1:ncol(datMy)], center=T, scale=T)
corMatMy <- cor(datMy.scale, method="spearman")
corrplot(corMatMy, order="hclust", tl.cex=1)

highlyCor <- findCorrelation(corMatMy, 0.80) #After inspection, apply correlation filter at 0.70,

#then we remove all the variable correlated with more 0.8.
datMyFiltered.scale <- datMy.scale[,-highlyCor]
corMatMy <- cor(datMyFiltered.scale)
corrplot(corMatMy, order = "hclust", tl.cex=1)

```

![alt_txt][BIOclim.all]
[BIOclim.all]:https://cloud.githubusercontent.com/assets/12142475/20832670/d86ef6c2-b88b-11e6-9ad7-e06544448f51.png

![alt_txt][BIOclim.reduced]
[BIOclim.reduced]:https://cloud.githubusercontent.com/assets/12142475/20832675/dde218c8-b88b-11e6-8158-0eabc1f40338.png





##RDA
Input files:

MAF of all loci

Geographic coordinates

Climate variables
	
	1. BIO5: Mat temp warmest month
	
	2. BIO15: Precipitation seasonality
	
	3. BIO13: Precipitation Wettest month
	
	4. BIO18: Precipitation warmest quarter (correlated with growth season)
	
	5. BIO2: Mean diurnal range

2081 loci


```
###1. MAF

#Calculate MAF for the full dataset within region using PLINK

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/RDA/

###Use the *nosex file to create a file for subsetting the data. Here I've made 1=within region, 2=within pop

plink --file SE132.FINAL.plink --freq --within SE132.region.cluster --out SE132.cluster

plink --file SE132.FINAL.plink --freq --within SE132.pop.cluster --out SE132.pop


###R
######Reformat PLINK output
###MAF for each locus -> melt and reformat rows as pops, and columns as loci.

SE.MAF <- read.table("SE132.pop.frq.strat", header=T)
SE.MAF2 <- SE.MAF[,c(3,2,6)]
summary(SE.MAF2)
         CLST               SNP             MAF        
 X1.Sk.Ho  : 2081   100865:23 :   15   Min.   :0.0000  
 X1.Sk.SF  : 2081   101108:100:   15   1st Qu.:0.0000  
 X1.Sk.SL  : 2081   101142:72 :   15   Median :0.1111  
 X2.Upp.Gra: 2081   101270:3  :   15   Mean   :0.2028  
 X2.Upp.K  : 2081   101367:84 :   15   3rd Qu.:0.3333  
 X2.Upp.O  : 2081   101609:30 :   15   Max.   :1.0000  
 (Other)   :18729   (Other)   :31125     

library("ggplot2")
library("reshape2")

SE.MAF3 <- melt(SE.MAF2, id.vars = c("CLST", "SNP"), variable_name = c("MAF"))
str(SE.MAF3)
head(SE.MAF3)


SE132.MAF4 <- dcast(SE.MAF3, formula= CLST ~ SNP)
head(SE145.MAF4)



##Add X infront of all locusnames. 
colnames(SE132.MAF4) <- paste("X", colnames(SE132.MAF4), sep=".")
write.csv(SE132.MAF4, file="SE.132.MAF.csv")
```


Run RDA

See this tutorial for the interpretation: REDUNDANCY ANALYSIS TUTORIAL: Landscape Genetics Paul Gugger redundancy-analysis-for-landscape-genetics.pdf on mac
```
library(vegan)

GenData <- read.csv("SE.132.MAF.csv", header=T)
GenData <- GenData[,15:2055]

Climate.Data <- read.csv("SE.145.MAF.csv", header=T)
names(Climate.Data)
Climate.variables <- Climate.Data[,1:14]
names(Climate.variables)
Climate.Data <- Climate.variables[,10:14]
Climate.Data$Lat <- Climate.variables$Lat
Climate.Data$Long <- Climate.variables$Long


##1. Run Full RDA model to determine how much of the variation is explainable by the expanatory variables we have
##H0: climate data does not affect genotype

RDA.SEfull <- rda(GenData ~ Lat + Long +bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled, Climate.Data) 
RDA.SEfull
Call: rda(formula = GenData ~ Lat + Long + bio5.scaled + bio15.scaled +
bio13.scaled + bio18.scaled + bio2.scaled, data = Climate.Data)

              Inertia Proportion Rank
Total         80.7640     1.0000     
Constrained   61.8140     0.7654    7
Unconstrained 18.9499     0.2346    7
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1   RDA2   RDA3   RDA4   RDA5   RDA6   RDA7 
25.969 13.331  7.957  4.612  4.274  3.625  2.046 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
5.153 3.555 2.913 2.256 1.959 1.732 1.381 

anova(RDA.SEfull)

Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ Lat + Long + bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled, data = Climate.Data)
         Df Variance      F Pr(>F)    
Model     7   68.333 3.2732  0.001 ***
Residual  7   20.876                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##to see which variables are most important, we can plot the results in a biplot

plot(RDA.SEfull)

```

![alt_txt][RDA.SEfull]
[RDA.SEfull]:https://cloud.githubusercontent.com/assets/12142475/20886240/61ca36c8-baf5-11e6-9b9d-420e099bac18.png



```
##Partial out geog
H0: Climate does not explain genetic data

pRDA.geog <- rda(GenData~bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled+ Condition(Lat + Long), Climate.Data)
head(summary(pRDA.geog))

anova(pRDA.geog)

Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled + Condition(Lat + Long), data = Climate.Data)
         Df Variance      F Pr(>F)   
Model     5   33.803 2.2668  0.003 **
Residual  7   20.876                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


H0 rejected: Climate does explain GeneticData

pRDA.geog
Call: rda(formula = GenData ~ bio5.scaled + bio15.scaled + bio13.scaled
+ bio18.scaled + bio2.scaled + Condition(Lat + Long), data =
Climate.Data)

              Inertia Proportion Rank
Total         89.2091     1.0000     
Conditional   34.5300     0.3871    2
Constrained   33.8026     0.3789    5
Unconstrained 20.8765     0.2340    7
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1   RDA2   RDA3   RDA4   RDA5 
15.549  9.094  4.488  2.712  1.960 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
6.190 3.872 3.460 2.250 2.179 1.662 1.264 


plot(pRDA.geog, main="pRDA (geog partialled out)")
```

![alt_txt][pRDA.geog]
[pRDA.geog]:https://cloud.githubusercontent.com/assets/12142475/20886394/22ed46a6-baf6-11e6-9d8b-41208c033d03.png


```
##Partial out climate

H0: Geog alone does not explain Genetic data

pRDA.climate <- rda(GenData~Lat+Long + Condition(bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled), Climate.Data)
pRDA.climate
Call: rda(formula = GenData ~ Lat + Long + Condition(bio5.scaled +
bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled), data =
Climate.Data)

               Inertia Proportion Rank
Total         89.20911    1.00000     
Conditional   60.53395    0.67856    5
Constrained    7.79869    0.08742    2
Unconstrained 20.87648    0.23402    7
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2 
5.364 2.435 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
6.190 3.872 3.460 2.250 2.179 1.662 1.264 


head(summary(pRDA.climate))

anova(pRDA.climate)

Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ Lat + Long + Condition(bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled), data = Climate.Data)
         Df Variance      F Pr(>F)
Model     2   7.7987 1.3075  0.173
Residual  7  20.8765          

H0: not rejected -> Geography alone does not explain GenData

plot(pRDA.climate, main="pRDA (climate partialled out)")
```

![alt_txt][pRDA.climate]
[pRDA.climate]:https://cloud.githubusercontent.com/assets/12142475/20886517/a00af9b2-baf6-11e6-85f8-10ed240ea369.png


Find the most important loci associated with Climate
```
 summary(pRDA.geog)
 
Aattccumulated constrained eigenvalues
Importance of components:
                       RDA1  RDA2   RDA3    RDA4    RDA5
Eigenvalue            15.55 9.094 4.4877 2.71164 1.96009
Proportion Explained   0.46 0.269 0.1328 0.08022 0.05799
Cumulative Proportion  0.46 0.729 0.8618 0.94201 1.00000

                 RDA1     RDA2    RDA3     RDA4     RDA5 PC1
bio5.scaled   0.10875 -0.16439  0.2365 -0.07146  0.10935   0
bio15.scaled  0.51115  0.23431  0.3214 -0.04714 -0.05866   0
bio13.scaled  0.47356 -0.23902 -0.3016 -0.17095 -0.53324   0
bio18.scaled  0.47439 -0.06841  0.2092  0.02536 -0.26091   0
bio2.scaled  -0.05063 -0.13750  0.2565 -0.06603 -0.02417   0

RDA1: BIO13, BIO15, BIO18

RDA2: BIO13, BIO15
```

##LFMM

See tutorial for command line version: http://membres-timc.imag.fr/Olivier.Francois/lfmm/files/LEA_1.html

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/LFMM/SE132.FINAL

Input files

1. .env environmental file

2. Genotype file
	
	2.1. 2081 loci (all loci)
	
	2.2. only loci variable in >5 populations


1. Env file
```
#
env <- read.table("SE132.norm.envvariables", header=F) ###read in the environmental data. One line perindividual. Same order as .vcf file
write.env(env, "SE133.env")   ##convert to correct .env format

```


2. Genotype file. Convert .vcf to .lfmm in LEA
```
library(LEA)
genotype = vcf2lfmm("SE132.FINAL.vcf")

	- number of detected individuals:	132
	- number of detected loci:		2081

For SNP info, please check ./SE145.FINAL.vcfsnp.

0 line(s) were removed because these are not SNPs.
Please, check ./SE145.FINAL.removed file, for more informations.

```


###Population structure with SNMF

The first step is to evaluate the population structure. SNMF works on the same principles as STRUCTURE, but is much faster. 

Test K1:10
```
obj.snmf = snmf(genotype, K=1:10, entropy=T, ploidy=2, project="new")

##
The project is saved into :
 SE132.FINAL.snmfProject 

To load the project, use:
 project = load.snmfProject("SE132.FINAL.snmfProject")

To remove the project, use:
 remove.snmfProject("SE132.FINAL.snmfProject")
```

Since the program was called with the entropy option, we can plot the values of the cross-entropy criterion for each K. The value for which the function plateaus or increases is our estimate of K

```
plot(obj.snmf)

```
![alt_txt][snmf2081]
[snmf2081]:https://cloud.githubusercontent.com/assets/12142475/20887077/6ce017a4-baf9-11e6-858f-d4918824192e.png




We observe that the minimum value is around K=5. We can also visualize a barplot of ancestry coeffcients as follows.

```
barplot(t(Q(obj.snmf, K=5)), col=1:5)
```

![alt_txt][barplot2041]
[barplot2041]:https://cloud.githubusercontent.com/assets/12142475/20887148/c74b3a7a-baf9-11e6-9cf1-709b181f71c7.png



###Run LFMM

on fgcz47 

```
scp * fgcz47:/srv/kenlab/alexjvr_p1795/SE.Analyses/LFMM.Dec2016/SE.2041.lfmm/

##R
obj.lfmm = lfmm("SE132.FINAL.lfmm", "SE132.env", K=5, rep=5, project="new")


export.lfmmProject(obj.lfmmProject)   ##this creates a .zip file in the current directory, which can easily be transferred to other folders. 

import.lfmmProject(obj.lfmmProject) ##to import the project into R from the new folder
```
move the project back onto the mac

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/LFMM/SE132.FINAL

And calculate the corrected p-values of all the loci. 

#####Genomic inflation (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137506/)

https://bioinformaticsngs.wordpress.com/2016/03/08/genomic-inflation-factor-calculation/

The genomic inflation factor λgc  is defined as the ratio of the median of the empirically observed distribution of the test statistic to the expected median, thus quantifying the extent of the bulk inflation and the excess false positive rate. In other words, λgc is defined as the median of the resulting chi-squared test statistics divided by the expected median of the chi-squared distribution. The median of a chi-squared distribution with one degree of freedom is 0.4549364 (qchisq(0.5,1) in R). A λgc value can be calculated from z-scores, chi-square statistics, or p-values, depending on the output you have from the association analysis.

Population structure, including population stratification and cryptic relatedness, can cause spurious associations in genome-wide association studies (GWAS). Usually, the scaled median or mean test statistic for association calculated from multiple single-nucleotide-polymorphisms across the genome is used to assess such effects, and ‘genomic control' can be applied subsequently to adjust test statistics at individual loci by a genomic inflation factor. Published GWAS have clearly shown that there are many loci underlying genetic variation for a wide range of complex diseases and traits, implying that a substantial proportion of the genome should show inflation of the test statistic. Here, we show by theory, simulation and analysis of data that in the absence of population structure and other technical artefacts, but in the presence of polygenic inheritance, substantial genomic inflation is expected. Its magnitude depends on sample size, heritability, linkage disequilibrium structure and the number of causal variants. Our predictions are consistent with empirical observations on height in independent samples of ∼4000 and ∼133 000 individuals

Normally we divide by the genomic inflation factor 0.456 to get lambda. Lambda should be close to 1, and the adj.p-value distribution should be flat. If this is not correct, then we can increase the genomic inflation factor to obtain a lambda closer to 1. (since the genomic inflation factor is known to be conservative). 
Here I have adjusted the values to 0.85. 

```
zs.d1 <- z.scores(SE132.lfmm, K=5, d=1)
zs.d1.median =apply(zs.d1, MARGIN=1, median)


lambda=median(zs.d1.median^2)/0.4549364
lambda
	1] 1.928976
lambda=median(zs.d1.median^2)/0.85
lambda
	[1] 1.034839
adj.p.values.d1 =pchisq(zs.d1.median^2/0.85, df=1, lower=F)


zs.d2 <- z.scores(SE132.lfmm, K=5, d=2)
zs.d2.median =apply(zs.d2, MARGIN=1, median)
lambda=median(zs.d2.median^2)/0.4549364
lambda
	[1] 1.946945
lambda=median(zs.d2.median^2)/0.85
lambda
	[1] 1.044479
adj.p.values.d2 =pchisq(zs.d2.median^2/0.85, df=1, lower=F)

zs.d3 <- z.scores(SE132.lfmm, K=5, d=3)
zs.d3.median =apply(zs.d3, MARGIN=1, median)
lambda=median(zs.d3.median^2)/0.0.4549364
lambda
	[1] 1.034471
	
lambda=median(zs.d3.median^2)/0.85
lambda
	[1] 0.5549635
adj.p.values.d3 =pchisq(zs.d3.median^2/0.0.4549364, df=1, lower=F)

zs.d4 <- z.scores(SE132.lfmm, K=5, d=4)
zs.d4.median =apply(zs.d4, MARGIN=1, median)
lambda=median(zs.d4.median^2)/0.0.4549364
lambda
	[1] 1.492845
lambda=median(zs.d4.median^2)/0.85
lambda
	[1] 0.8008673
adj.p.values.d4 =pchisq(zs.d4.median^2/0.85, df=1, lower=F)

zs.d5 <- z.scores(SE132.lfmm, K=5, d=5)
zs.d5.median =apply(zs.d5, MARGIN=1, median)
lambda=median(zs.d5.median^2)/0.0.4549364
lambda
	[1] 2.888479
lambda=median(zs.d5.median^2)/0.85
lambda
	[1] 1.034839
adj.p.values.d5 =pchisq(zs.d5.median^2/0.85, df=1, lower=F)


par(mfrow=c(3,2))
hist(adj.p.values.d1)
hist(adj.p.values.d2)
hist(adj.p.values.d3)
hist(adj.p.values.d4)
hist(adj.p.values.d5)



```





##Overlap between Fst chart and outliers

plot the histogram of Fst for the outliers identified with 1)RDA, and 2) LFMM, on the full Fst plot to look determine whether the same loci are outliers. 

Run all outlier analyses (RDA & LFMM) with 2 datasets: 1) full dataset, 2) reduced dataset including only loci variable in >5 populations. 

```
##Get the overall per locus Fst using R
#the genind object would've been read in earlier for the sum stats. If not, convert .ped to .str format and import into R
library(adegenet)
library(hierfstat)

#SE132.genind <- read.structure("SE132all.str")

hier.SE132 <- genind2hierfstat(SE132.genind)
stats.SE132 <- basic.stats(hier.SE132)
stats.SE132.perlocus <- stats.SE132$perloc

##Linux
nano Fstoutliers.names  ##paste the locus names of all the outliers identified by different methods

##R

#Change the locus names if necessary: 
rownames(stats.SE132.perlocus) <- sub("X", "X.", rownames(stats.SE132.perlocus))
fstoutliers.all <- read.table("Fstoutlier.names", header=T)
names(fstoutliers.all)

RDA1.list <- as.character(fstoutliers.all$RDA1)
RDA2.list <- as.character(fstoutliers.all$RDA2)

RDA1.fst <- stats.SE132.perlocus[RDA1.list,]
RDA2.fst <- stats.SE132.perlocus[RDA2.list,]

RDA2.fst <- RDA2.fst$Fst
RDA1.fst <- RDA1.fst$Fst

fstall.hist <- hist(stats.SE132.perlocus$Fst)
RDA1.hist <- hist(RDA1.fst)
RDA2.hist <- hist(RDA2.fst)
plot(fstall.hist, xlim=c(-0.15,1.0), breaks=c(115))
plot(RDA1.hist, col=rgb(0,0,1,1/4), xlim=c(-0.15,1.0), breaks=c(115), add=T)
plot(RDA2.hist, col=rgb(1,0,0,1/4), xlim=c(-0.15,1.0), breaks=c(115), add=T)

```

![alt_txt][Fst.outliers]
[Fst.outliers]:https://cloud.githubusercontent.com/assets/12142475/20971473/d786eb8e-bc91-11e6-897e-40c15fade58b.png



##IBD vs IBE using EEMS

Looking for deviations from IBE across the landscape

https://github.com/dipetkov/eems/blob/master/Documentation/EEMS-doc.pdf

3 input files needed

pairwise genetic distance

polygon of coordinates

coords file.

1. pairwise genetic distance as calculated with bed2diffs

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/EEMS

```
/Users/alexjvr/Applications/eems-master/bed2diffs/src-wout-openmp/bed2diffs_v1 --bfile SE132.FINAL.plink

```

2. Polygon of coordinates. (anticlockwise. First and last coordinates should be the same

Can be obtained here:

http://www.birdtheme.org/useful/v3tool.html


3. Coords file

Coordinates for each individual. One line per sample. Two columns.

Same order as in the plink file.


Run the simulations from the EEMS directory: /Users/alexjvr/Applications/eems-master/runeems_snps/src

First, create 3 parameter files  (see BV.params-chain1.ini)

Then run all three chains
```
./runeems_snps --params BV.params-chain1.ini --seed 123

```

Launch R in directory where results are deposited
```
install.packages("rEEMSplots", repos=NULL, type="source")
library(rEEMSplots)

eems.results.to.plot = paste(path.package("rEEMSplots"), "/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/EEMS/SE.EEMS.plots", sep="")

name.figures.to.save="SE132.rEEMSplots"
eems.plots(mcmcpath="./SE132.chain1", plotpath=paste(name.figures.to.save, sep=""), longlat=T)

```

##RandomForest Model

Pool all of the outliers and map them using the RandomForest model. 




#Repeat everything with Dataset2

Create a second dataset which includes only loci that are variable in >5 populations. 

1. Filter the dataset

2. calculate per locus Fst and plot

3. RDA with Dataset2

4. LFMM with Dataset2

5. Fst plot with outliers overlaid

6. Compare Dataset1 & Dataset2 outliers

7. 


```
##R
setwd("/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/SE132.FINAL/SE132.FINAL.subset.data")

frq.var.files <- list.files(pattern="frq$")
frq.var.files
frq.var.files.list <- lapply(frq.var.files, read.table, header=T)
names(frq.var.files.list) <- frq.var.files
names(frq.var.files.list)

attach(frq.var.files.list)
frq.var.frequency.keep <- do.call(rbind, lapply(names(frq.var.files.list), get))
summary(frq.var.frequency)
frq.var.frequency.keep <- data.frame(table(frq.var.frequency$SNP))
hist(frq.var.frequency.keep$Freq, xlab="Nr pops", ylab="Frequency of loci across 15 pops")


frq.var.files.list <- lapply(frq.var.files.list, subset, MAF>0.01)  ##keep only the variable loci
summary(frq.var.files.list)
attach(frq.var.files.list)
frq.var.frequency <- do.call(rbind, lapply(names(frq.var.files.list), get))
summary(frq.var.files.list)
frq.var.frequency.keep <- data.frame(table(frq.var.frequency$SNP))
summary(frq.var.frequency.keep)

frq.var.frequency.keep <- subset(frq.var.frequency.keep, Freq>5)
hist(frq.var.frequency.keep$Freq, xlab="Nr pops", ylab="Frequency of loci across 15 pops")


write.table(frq.var.frequency.keep, "loci.var5pops.names", quote=F, row.names=F)
```

2081 loci 

![alt_txt][SE132.FINAL.variable.loci]
[SE132.FINAL.variable.loci]:https://cloud.githubusercontent.com/assets/12142475/20984944/9923e35c-bcc2-11e6-9461-4724704bbc4f.png

1707 loci variable in >5 populations

![alt_txt][SE132.FINAL.variable.loci]
[SE132.FINAL.variable.loci]:https://cloud.githubusercontent.com/assets/12142475/20984990/c8891a2c-bcc2-11e6-8157-ae4e878343c4.png


subset the dataset
```
vcftools --vcf SE132.FINAL.vcf --snps SE132.FINAL.subset.data/loci.var5pops.names --recode --recode-INFO-all --out SE132.1707.FINAL

vcftools --vcf SE132.1707.FINAL.vcf --plink --out SE132.1707.plink
plink --file SE132.1707.plink --recode --recodeA --out SE132.1707.plink
```

Sumstats Dataset2

```
##linux
mkdir Dataset2/SumSats
cp rawdataFilter/SE132.FINAL/SE132.1707* Dataset2/SumSats/

##convert to structure using pgdspider

##R

library(adegenet)
library(hierfstat)

SE.data2 <- read.structure("SE132.1707.str")

 How many genotypes are there? 132

 How many markers are there? 1707

data2.names <- read.table("SE.data2.names", header=T)
SE.data2.genind <- SE.data2

data2.pop <- as.factor(data2.names$pop.name)
SE.data2.genind@pop <- data2.pop
SE.data2.genind@pop

hier.SEdata2 <- genind2hierfstat(SE.data2.genind)
stats.hier.SEdata2 <- basic.stats(hier.SEdata2)
stats.hier.SEdata2.perloc <- stats.hier.SEdata2$perloc

summary(stats.hier.SEdata2.perloc$Fst)
hist(stats.hier.SEdata2.perloc$Fst, xlim=c(-0.1, 1.0), breaks=110)
```

Dataset2 Fst distribution 

![alt_txt][Data2.Fst]
[Data2.Fst]:https://cloud.githubusercontent.com/assets/12142475/20985654/a7e330c0-bcc5-11e6-9a1b-b9f587477757.png


###RDA: Dataset2


```


```



