#SE: Final dataset

After speaking with Victoria, I will include all the populations (even with <5 individuals per population) in the final dataset. Also, based on population structure analyses, I will exclude the DE populations, as they are from a different gene pool. 

##Step 1: Remove DE

```
vcftools --vcf SEFinalc94d6m4p3.vcf --recode --recode-INFO-all --out SEFinalc94d6m4p3.vcf
bcftools query -l SEFinalc94d6m4p3.vcf

bcftools reheader SEFinalc94d6m4p3.vcf.recode.vcf -s raw.SE193.names --out SE193.raw.vcf

vcftools --vcf SE193.raw.vcf --remove DE.names --recode --recode-INFO-all --out SE163
```

##Step 2: Remove loci genotyped in <80% individuals

```
vcftools --vcf SE163.recode.vcf --max-missing 0.8 --recode --recode-INFO-all --out SE163.s1

After filtering, kept 16640 out of a possible 1300946 Sites
```

Look at missingness
```
vcftools --vcf SE163.s1.recode.vcf --missing-indv

##R
library(ggplot2)
SE.163.s1 <- read.table("out.imiss", header=T)
pop <- read.table("SE163.popnames", header=T)
SE.163.s1$pop <- pop$pop
SE.163.s1$pop.order <- pop$pop.order

SE.163.s1.sort <- SE.163.s1[order(SE.163.s1$pop.order),]

SE.163.s1.sort$pop <- factor(SE.163.s1.sort$pop, levels=SE.163.s1.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.163.s1.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][s1.missing]
[s1.missing]:https://cloud.githubusercontent.com/assets/12142475/20754370/dd1ec17a-b70a-11e6-977d-08cf89dd767e.png


Calculate SFS & LD for s1
```
vcftools --vcf SE163.s1.recode.vcf --plink --out SE163.s1.plink 
plink --file SE163.s1.plink --recode --recodeA --out SE163.plink

plink --file SE163.s1.plink --freq --out SE163.s1.plink
plink --file SE163.s1.plink --r2 --out SE163.s1.plink
```


##Step 3-5 MAF, HWE, LD, thin

After each step, calculate LD & SFS
```
vcftools --vcf SE163.s1.recode.vcf --maf 0.05 --recode --recode-INFO-all --out SE163.s2

After filtering, kept 3948 out of a possible 16640 Sites

vcftools --vcf SE163.s2.recode.vcf --plink --out SE163.s2.plink 
plink --file SE163.s2.plink --recode --recodeA --out SE163.s2.plink

plink --file SE163.s2.plink --freq --out SE163.s2.plink
plink --file SE163.s2.plink --r2 --out SE163.s2.plink
```

Calculate HWE and LD
```
mkdir popnames.plink.folder ##directory with 15 files each containing the population names
mkdir subset.data
for i in $(ls popnames.plink.folder/); do plink --file SE163.s2.plink --keep popnames.plink.folder/$i --recode --recodeA --out subset.data/$i.plink; done

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --freq --out subset.data/$i; done
for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --r2 --out subset.data/$i; done

##R
Sk.Ho.frq <- read.table("subset.data/Sk.Ho.frq", header=T)
Sk.SF.frq <- read.table("subset.data/Sk.SF.frq", header=T)
Sk.SL.frq <- read.table("subset.data/Sk.SL.frq", header=T)

Upp.Gra.frq <- read.table("subset.data/Upp.Gra.frq", header=T)
Upp.K.frq <- read.table("subset.data/Upp.K.frq", header=T)
Upp.O.frq <- read.table("subset.data/Upp.O.frq", header=T)

Um.Taf.frq <- read.table("subset.data/Um.Taf.frq", header=T)
Um.Gr.frq <- read.table("subset.data/Um.Gr.frq", header=T)
Um.UM3.frq <- read.table("subset.data/Um.UM3.frq", header=T)

LT1.frq <- read.table("subset.data/LT1.frq", header=T)
LT2.frq <- read.table("subset.data/LT2.frq", header=T)
LT3.frq <- read.table("subset.data/LT3.frq", header=T)

Kir.L.frq <- read.table("subset.data/Kir.L.frq", header=T)
Kir.G.frq <- read.table("subset.data/Kir.G.frq", header=T)

FIN.frq <- read.table("subset.data/FIN.frq", header=T)

par(mfrow=c(5,3))
hist(Sk.Ho.frq$MAF, main="Sk.Ho (20, 3918, 81.6% genotyping) SFS")
hist(Sk.SF.frq$MAF, main="Sk.SF (10, 3918, 95.8%) SFS")
hist(Sk.SL.frq$MAF, main="Sk.SL (19, 3918, 91.3%) SFS")

hist(Upp.Gra.frq$MAF, main="Upp.Gra (10, 3918, 86.5%) SFS")
hist(Upp.K.frq$MAF, main="Upp.K (10, 3918, 96.3%) SFS")
hist(Upp.O.frq$MAF, main="Upp.O (10, 3918, 84.9%) SFS")

hist(Um.UM3.frq$MAF, main="Umea.UT3 (4, 3918, 55%) SFS")
hist(Um.Taf.frq$MAF, main="Umea.Taf (10, 3918, 92.3%) SFS")
hist(Um.Gr.frq$MAF, main="Umea.Gr (10, 3918, 90.9%) SFS")

hist(LT1.frq$MAF, main="LT1 (10, 3918, 92.5%) SFS")
hist(LT2.frq$MAF, main="LT2 (10, 3918, 58.2%) SFS")
hist(LT3.frq$MAF, main="LT3 (10, 3918, 80.4%) SFS")

hist(Kir.G.frq$MAF, main="Kir.G (10, 3918, 96.3%) SFS")
hist(Kir.L.frq$MAF, main="Kir.L (10, 3918, 94.7%) SFS")

hist(FIN.frq$MAF, main="FIN (10, 3918, 75.4%) SFS")
```

![alt_txt][SFS.s2]
[SFS.s2]:https://cloud.githubusercontent.com/assets/12142475/20756148/dd752716-b711-11e6-84bb-20cad8635e76.png

Only variable loci
```
#Create a list of all the .frq dataframes
SE.s2.freq.list <- setNames(lapply(ls(pattern=".frq"), function(x) get(x)), ls(pattern=".frq"))

#and remove all .frq from the global env
rm(list=ls(pattern=".frq"))

#Filter all for MAF>0.001
SE.s2.freq.var.list <- lapply(SE.s2.freq.list, subset, MAF>0.01)

#plot hist of all new MAF with title
par(mfrow=c(3,5))
for(i in names(SE.s2.freq.var.list))
{
    df1 = as.data.frame(SE.s2.freq.var.list[[i]])
    hist(df1$MAF, main=i)
}

```

![alt_txt][SFS.s2]
[SFS.s2]:https://cloud.githubusercontent.com/assets/12142475/20760255/c883f4aa-b71f-11e6-97c9-0aea039f5bf6.png

Variable loci across all populations
```
attach(SE.s2.freq.var.list)
SE.s2.var.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.var.list), get))
summary(SE.s2.var.loci.freq)

SE.s2.var.loci.freq.keep <- data.frame(table(SE.s2.var.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.var.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of variable loci across 15 pops", breaks=seq(0.5,15.5, by=1.0))
detach(SE.s2.freq.var.list)
```

![alt_txt][variable.s2]
[variable.s2]:https://cloud.githubusercontent.com/assets/12142475/20762249/09d0d094-b726-11e6-8946-1814e6810cad.png

Fixed loci across all populations
```
#Filter all for fixed loci
SE.s2.freq.fixed.list <- lapply(SE.s2.freq.list, subset, MAF<0.000001)

attach(SE.s2.freq.fixed.list)
SE.s2.fixed.loci.freq <- do.call(rbind, lapply(names(SE.s2.freq.fixed.list), get))
summary(SE.s2.fixed.loci.freq)

SE.s2.fixed.loci.freq.keep <- data.frame(table(SE.s2.fixed.loci.freq$SNP)) 
#SE.var.loci.freq.keep <- subset(SE.var.loci.freq.keep, Freq>0) 
hist(SE.s2.fixed.loci.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of fixed loci across 15 pops", breaks=seq(0,14, by=1.0))
detach(SE.s2.freq.fixed.list)
```
![alt_txt][fixed.s2]
[fixed.s2]:https://cloud.githubusercontent.com/assets/12142475/20762566/1249f51a-b727-11e6-909f-99a857a6a39d.png





##step 6: Final checks: HWE, LD, missingness

LD
```
#R write tables of fixed loci
##Actually I can just filter for MAF < 0.005, so I've changed this code
for(i in names(SE.s2.freq.fixed.list)){
write.table(SE.s2.freq.fixed.list[[i]][2], paste(i,".fixedloci", sep=""), row.names=F, col.names=F, quote=F)
}
##linux
#for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --exclude $i.frq.fixedloci --recode --recodeA --out ld.subset.data/$i.var.plink; done



##new code
##LINUX
mkdir ld.subset.data

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --maf 0.005 --recode --recodeA --out ld.subset.data/$i.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file subset.data/$i.plink --r2 --recode --recodeA --out ld.subset.data/$i.var; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
ld.subset.path <- './ld.subset.data/'
ld.var.files <- list.files(ld.subset.path, pattern='ld$')
ld.var.files.list <- lapply(ld.var.files, read.table, header=T)
names(ld.var.files.list) <- ld.var.files
names(ld.var.files.list)

##plot ld for each population
for(i in names(ld.var.files.list))
{
    df1 = as.data.frame(ld.var.files.list[[i]])
    hist(df1$R2, main=i)
}

#Calculate frequency lf LD
SE.s3.ld.R2.08.list <- lapply(ld.var.files.list, subset, R2>0.8)
attach(SE.s3.ld.R2.08.list)
SE.s3.ld.R2.08.freq <- do.call(rbind, lapply(names(SE.s3.ld.R2.08.list), get))
summary(SE.s3.ld.R2.08.freq)

SE.s3.ld.R2.08.freq.keep <- data.frame(table(SE.s3.ld.R2.08.freq$SNP_A, SE.s3.ld.R2.08.freq$SNP_B)) 
summary(SE.s3.ld.R2.08.freq.keep)
hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))
#SE.s3.ld.R2.08.freq.keep <- subset(SE.s3.ld.R2.08.freq.keep, Freq>0) 
#hist(SE.s3.ld.R2.08.freq.keep$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of linkage (R2>0.8) across 15 pops", breaks=seq(0,14, by=1.0))

```

![alt_txt][LD.all]
[LD.all]:https://cloud.githubusercontent.com/assets/12142475/20789034/49ad8328-b7b3-11e6-8e6c-e49de3f02883.png

![alt_txt][LD.freq]
[LD.freq]:https://cloud.githubusercontent.com/assets/12142475/20790452/a8f614da-b7b8-11e6-8bf4-3e414fce13d9.png


So LD between loci are mostly only in 1 population. The max is 6 (1 locus), and then 4 (6 loci). So I will not filter any of these loci out. 


HWE
Make sure that all the missing data are also filtered out of the datasets. Using the old script above, only the fixed loci per population are removed, but the ungenotyped loci remain. So there will be gaps in the .hwe file, and this can't be imported into R. 
```
##Linux
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --maf 0.005 --out ld.subset.data/$ivar.plink; done
for i in $(ls popnames.plink.folder/); do plink --file ld.subset.data/$i.var.plink --hardy --out ld.subset.data/$i; done

##R
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
hwe.subset.path <- './ld.subset.data/'
hwe.files <- list.files(hwe.subset.path, pattern='hwe$')
hwe.files
setwd('./ld.subset.data')
hwe.files.list <- lapply(hwe.files, read.table, header=T)
names(hwe.files.list) <- hwe.files  ##name each dataframe

#subset the data to include only TEST=="ALL"
hwe.files.list <- lapply(hwe.files.list, subset, TEST=="ALL")

#create a list of df with all loci out of HWE and O.Het>0.5
hwe.files.list.remove <- lapply(hwe.files.list, subset, (P<0.050001|O.HET.>0.5))

attach(hwe.files.list.remove)
hwe.remove.freq <- do.call(rbind, lapply(names(hwe.files.list.remove), get))
summary(hwe.remove.freq)
detach(hwe.files.list.remove)

hwe.remove.freq.freq <- data.frame(table(hwe.remove.freq$SNP)) 
summary(hwe.remove.freq.freq)
hist(hwe.remove.freq.freq$Freq, xlab="Nr pops", ylab="Frequency", main="Frequency of loci deviating from HWE and O.Het>0.5 across 15 pops"))

```

![alt_txt][HWE.freq]
[HWE.freq]:https://cloud.githubusercontent.com/assets/12142475/20799028/32c4e7fe-b7e1-11e6-868c-4b7d2fca003a.png


137 Loci to remove
```
#R
HWE.loc.remove <- subset(hwe.remove.freq.freq, Freq>5)
summary(HWE.loc.remove)

setwd("/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter")
write.table(HWE.loc.remove$Var1, "HWE.loc.remove", col.name=F, row.name=F, quote=F)


#Linux
plink --file SE163.s3.plink --exclude HWE.loc.remove --recode --recodeA --out SE163.s4.plink
```
Convert to vcf using pgdspider





Missingness

```
vcftools --vcf SE163.s4.vcf --recode --recode-INFO-all --out SE163.s4
vcftools --vcf SE163.s4.recode.vcf --missing-indv

##R
library(ggplot2)
SE.163.s4 <- read.table("out.imiss", header=T)
pop <- read.table("SE163.popnames", header=T)
SE.163.s4$pop <- pop$pop
SE.163.s4$pop.order <- pop$pop.order

SE.163.s4.sort <- SE.163.s4[order(SE.163.s4$pop.order),]

SE.163.s4.sort$pop <- factor(SE.163.s4.sort$pop, levels=SE.163.s4.sort$pop)   ##sort pop.nr. Numbers from south to North

qplot(pop, F_MISS, data=SE.163.s4.sort, geom=c("boxplot", "jitter"))
```

![alt_txt][missing.s4]
[missing.s4]:https://cloud.githubusercontent.com/assets/12142475/20800272/3763c2ea-b7e5-11e6-9cba-30385821345c.png


remove individuals with >40% missingness
```
SE163.imiss.40 <- subset(SE.163.s4, F_MISS>0.4)
summary(SE163.imiss.40)
write.table(SE163.imiss.40$INDV, "SE163.imiss40", col.name=F, row.name=F, quote=F)

#LINUX
vcftools --vcf SE163.s4.recode.vcf --remove SE163.imiss40 --recode --recode-INFO-all --out SE145.FINAL

##rename individuals
bcftools reheader SE145.FINAL.recode.vcf -s names.SE145FINAL --out SE145.FINAL.vcf

#convert to plink
vcftools --vcf SE145.FINAL.vcf --plink --out SE145.FINAL.plink
plink --file SE145.FINAL.plink --recode --recodeA --out SE145.FINAL.plink

##subset into populations
mkdir FINAL.subset.data

for i in $(ls popnames.plink.folder/); do plink --file SE145.FINAL.plink --keep popnames.plink.folder/$i --recode --recodeA --out FINAL.subset.data/$i.FINAL.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.plink --maf 0.05 --recode --recodeA --out FINAL.subset.data/$i.FINAL.var.plink; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --r2 --out FINAL.subset.data/$i.FINAL.var; done

for i in $(ls popnames.plink.folder/); do plink --file FINAL.subset.data/$i.FINAL.var.plink --freq --out FINAL.subset.data/$i.FINAL.var; done


##Final graphs of per population SFS and LD
##LD
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.ld.files <- list.files(FINAL.subset.path, pattern='ld$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/FINAL.subset.data/')
FINAL.ld.files.list <- lapply(FINAL.ld.files, read.table, header=T)
names(FINAL.ld.files.list) <- FINAL.ld.files
names(FINAL.ld.files.list)

##plot ld for each population
par(mfrow=c(5,3))
for(i in names(FINAL.ld.files.list))
{
    df1 = as.data.frame(FINAL.ld.files.list[[i]])
    hist(df1$R2, main=i)
}

##SFS
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter')
FINAL.subset.path <- './FINAL.subset.data/'
FINAL.sfs.files <- list.files(FINAL.subset.path, pattern='var.frq$')
setwd('/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/rawdataFilter/FINAL.subset.data/')
FINAL.sfs.files.list <- lapply(FINAL.sfs.files, read.table, header=T)
names(FINAL.sfs.files.list) <- FINAL.sfs.files
names(FINAL.sfs.files.list)

##plot ld for each population
par(mfrow=c(5,3))
for(i in names(FINAL.sfs.files.list))
{
    df1 = as.data.frame(FINAL.sfs.files.list[[i]])
    hist(df1$MAF, main=i)
}


```

![alt_txt][FINAL.ld.perpop]
[FINAL.ld.perpop]:https://cloud.githubusercontent.com/assets/12142475/20803450/8dd6b0b0-b7ef-11e6-9b45-23215e6c70f6.png

![alt_txt][FINAL.sfs.perpop]
[FINAL.sfs.perpop]:https://cloud.githubusercontent.com/assets/12142475/20803455/92c8d9fe-b7ef-11e6-93e7-fcb1e06a0aed.png

####FINAL DATA

SE145.FINAL.plink /.vcf

145 individuals

2041 loci

15 populations

6 regions


##Population Structure

###Fst

Convert plink FINAL to structure using pgdspider, and import into R
```
/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/SumStats

library(adegenet)
library(hierfstat)
library(reshape)

SEall.145 <- read.structure("SE145.FINAL.str")
SEall.145

pop.SE145 <- read.table("popnames.SE145.structure", header=T)  ##make sure the populations are numbered "X1.Sk.Ho", etc.
pop.factor <- as.factor(pop.SE145$pop)
SEall.145@pop <- pop.factor

hier.SEall <- genind2hierfstat(SEall.145)

SEall.fst <- pairwise.fst(SEall.145, pop=NULL, res.type=c("dist", "matrix"))

m <- as.matrix(SEall.fst)
m2 <- melt(m)[melt(upper.tri(m))$value,]
names(m2)<- c("c1","c2", "distance")

library(gplots)

shadesOfGrey <- colorRampPalette(c("grey100", "grey0"))  ##define the colourpalette. 

Dend <- read.table("heatmap.popcolours", header=T)  ##list of colour names for each population based on R colour palatte. In alphabetical order (as in genind file)
Dend.Colours <- as.character(Dend$colours.pop)

par(oma=c(1,1,2,1))
heatmap.2(as.matrix(SEall.fst), trace="none", RowSideColors=Dend.Colours, ColSideColors=Dend.Colours, col=shadesOfGrey, labRow=F, labCol=F, key.ylab=NA, key.xlab=NA, key.title="Fst Colour Key", keysize=0.9, main="Pairwise Fst of SEall: 15pops, 6regions, 2041loci")  ##RowSideColors is for the dendrogram on the row, ColSideColors for the upper dendrogram. Colour order should be the same as the input. The pop order is alphabetical in the output. 
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")

popnames.all <- as.character(Dend$pop)
legend("bottom", popnames.all, xpd = TRUE, horiz = TRUE, inset = c(0, 0), bty="o", pch=15, col=Dend.Colours, title="Side Dendrogram:Region")
```

![alt_txt][Fst.SEall]
[Fst.SEall]:https://cloud.githubusercontent.com/assets/12142475/20813395/bee82366-b815-11e6-8985-3a766d77e3fc.png


###IBD

Fst/(1-Fst) vs log.dist(km) - according to Rousset et al. 1997, this is correlated with the effective population density (De) and effective dispersal distance (variance)

```
##Fst 15 pops -> Fst/(1-Fst)

library(reshape)
library(fields)


m <- as.matrix(SEall.fst)
m
m2 <- melt(m)[melt(upper.tri(m))$value,]
names(m2) <- c("c1", "c2", "distance")
m2
m2$IBD <- m2$distance/(1-m2$distance)


SE.pop.coords <- read.table("SE.coords.15pop", header=T)
SEpop_lon.lat <- cbind(SE.pop.coords$Long, SE.pop.coords$Lat)
distance.matrix.SEpop <- rdist.earth(SEpop_lon.lat, miles=F)  ##great circle dist based on the coordinates
m.dist <- as.matrix(distance.matrix.SEpop)
summary(m.dist)

m2.dist <- melt(m.dist)[melt(upper.tri(m.dist))$value,]
names(m2.dist) <- c("c1", "c2", "distance")
summary(m2.dist)
m2.dist$log.km <- log(m2.dist$distance)


library(MASS)
#dens <- kde2d(m2$IBD,m2.dist$log.km, n=10)
#myPal <- colorRampPalette(c("white","blue","gold", "orange", "red"))
plot(m2$IBD~m2.dist$log.km, pch=20,cex=.5, xlab="log Geographic distance (km)", ylab="Fst/(1-Fst)")
#image(dens, col=transp(myPal(10),.7), add=TRUE)
abline(fit <- lm(m2$IBD~m2.dist$log.km))
legend("bottomright", bty="n", legend=paste("R2 =", format(summary(fit)$adj.r.squared, digits=4)))  ##and paste R2
title("Isolation by distance plot - SEall")
```

![alt_txt][IBD.SEall]
[IBD.SEall]:https://cloud.githubusercontent.com/assets/12142475/20813679/ce0aaa0c-b816-11e6-9b73-3cb1e9977874.png



###AMOVA

http://grunwaldlab.github.io/Population_Genetics_in_R/AMOVA.html

```
library(adegenet)
library(poppr)

SE145.strata <- pop.SE145[,1:3]  ##from text file. each column has one hierarchy level specified for all individuals. (indiv, pop, region)
SEall.145.genind <- SEall.145
SEall.145.genind@other <- SE145.strata ##renamed the SEall.145 genind to a better name

strata(SEall.145.genind) <- other(SEall.145.genind)

SEall.145.genclone <- as.genclone(SEall.145.genind)

SE145.amova <- poppr.amova(SEall.145.genclone, ~region/pop)

SE145.amova 
$call
ade4::amova(samples = xtab, distances = xdist, structures = xstruct)

$results
                            Df    Sum Sq   Mean Sq
Between region               5  4208.303 841.66051
Between pop Within region    9  1692.362 188.04026
Between samples Within pop 130  8955.771  68.89055
Within samples             145  7428.309  51.22972
Total                      289 22284.745  77.10984

$componentsofcovariance
                                           Sigma          %
Variations  Between region             14.149482  17.555900
Variations  Between pop Within region   6.387117   7.924785
Variations  Between samples Within pop  8.830415  10.956294
Variations  Within samples             51.229717  63.563021
Total variations                       80.596731 100.000000

$statphi
                         Phi
Phi-samples-total 0.36436979
Phi-samples-pop   0.14702623
Phi-pop-region    0.09612313
Phi-region-total  0.17555900

SE145.amovatest <- randtest(SE145.amova, nrepet=999)

SE145.amovatest

class: krandtest 
Monte-Carlo tests
Call: randtest.amova(xtest = SE145.amova, nrepet = 999)

Number of tests:   4 

Adjustment method for multiple comparisons:   none 
Permutation number:   999 
                        Test       Obs    Std.Obs   Alter Pvalue
1  Variations within samples 51.229717 -19.400051    less  0.001
2 Variations between samples  8.830415   5.382762 greater  0.001
3     Variations between pop  6.387117   8.494544 greater  0.001
4  Variations between region 14.149482   5.061908    less  1.000


##Randomised samples
SE145.random <- SEall.145.genclone
set.seed(9001)
strata(SE145.random) <- strata(SEall.145.genclone)[sample(nInd(SEall.145.genclone)), -1]
head(strata(SE145.random))
head(strata(SEall.145.genclone))
SE145.random.amova <- poppr.amova(SE145.random, ~region/pop)

SE145.random.amova   ##now all the variation is within samples and within populations. So no population structure evident. 
$call
ade4::amova(samples = xtab, distances = xdist, structures = xstruct)

$results
                            Df     Sum Sq   Mean Sq
Between region               5   467.1205  93.42409
Between pop Within region    9   806.6388  89.62653
Between samples Within pop 130 13582.6767 104.48213
Within samples             145  7428.3090  51.22972
Total                      289 22284.7449  77.10984

$componentsofcovariance
                                            Sigma           %
Variations  Between region              0.1012924   0.1312743
Variations  Between pop Within region  -0.7963466  -1.0320601
Variations  Between samples Within pop 26.6262058  34.5073950
Variations  Within samples             51.2297172  66.3933908
Total variations                       77.1608688 100.0000000

$statphi
                           Phi
Phi-samples-total  0.336066092
Phi-samples-pop    0.341993323
Phi-pop-region    -0.010334168
Phi-region-total   0.001312743


SE145.random.amovatest<- randtest(SE145.random.amova, nrepet=999) 

SE145.random.amovatest

class: krandtest 
Monte-Carlo tests
Call: randtest.amova(xtest = SE145.random.amova, nrepet = 999)

Number of tests:   4 

Adjustment method for multiple comparisons:   none 
Permutation number:   999 
                        Test        Obs     Std.Obs   Alter Pvalue
1  Variations within samples 51.2297172 -19.1916804    less  0.001
2 Variations between samples 26.6262058  17.6058635 greater  0.001
3     Variations between pop -0.7963466  -1.1612015 greater  0.893
4  Variations between region  0.1012924  -0.0216662    less  0.512

##plot both outputs
par(mfrow=c(2,1))
plot(SE145.amovatest)
plot(SE145.random.amovatest)
```

![alt_txt][AMOVA.SE145]
[AMOVA.SE145]:https://cloud.githubusercontent.com/assets/12142475/20814828/92bb3688-b81b-11e6-9c56-0a4d3123e3cb.png

![alt_txt][AMOVA.random]
[AMOVA.random]:https://cloud.githubusercontent.com/assets/12142475/20814833/9a1a0e36-b81b-11e6-9dd0-8b3ba47e0888.png



###DAPC

tutorial-dapc: A tutorial for Discriminant Analysis of Principal Components (DAPC) using adegenet 2.0.0

total variance = (variance between groups) + (variance within groups)

or more simply, denoting X the data matrix:

VAR(X) = B(X) +W(X)

Usual approaches such as Principal Component Analysis (PCA) or Principal Coordinates Analysis (PCoA / MDS) focus on V AR(X). That is, they only describe the global diversity, possibly overlooking differences between groups. On the contrary, DAPC optimizes B(X) while minimizing W(X): it seeks synthetic variables, the discriminant functions, which show differences between groups as best as possible while minimizing variation within clusters.

```
##1. estimate the number of clusters

Using k-means. Which finds the number of clusters with minimises W(X) and maximises B(X). Compare using BIC

Run algorithm on PCA transformed data. I.e. reduce the dataset so that it can run faster. 

grp.SE145 <- find.clusters(SEall.145.genind, max.n.clust=40)

> choose nr of PCs: 200  ##I try to keep all the PCs

> choose k: 4 ##i used K4-6. See table below


names.15 <- c("Sk.Ho", "Sk.SF", "SK.SL", "Upp.Gra", "Upp.K", "Upp.O", "Um.Gr", "Um.Taf", "Um.UT3", "LT1", "LT2", "LT3", "Kir.G", "Kir.L", "FIN")
names.15 <- as.character(names.15)
table.value(table(pop(SEall.145.genind), grp.SE145$grp), col.lab=paste("inf", 1:6), row.lab=names.15)

dapc1.SE145 <- dapc(SEall.145.genind, grp.SE145$grp)
scatter(dapc1.SE145)
```

Nr of PCs: I kept 200

![alt_txt][PCs.initial]
[PCs.initial]:https://cloud.githubusercontent.com/assets/12142475/20814979/3aefc648-b81c-11e6-98f2-18047b49a13c.png

Choose K: 4
![alt_txt][K.initial]
[K.initial]:https://cloud.githubusercontent.com/assets/12142475/20814984/3f5be07c-b81c-11e6-955d-6d169df56201.png


![alt_txt][K4]
[K4]:https://cloud.githubusercontent.com/assets/12142475/20815159/11a2dc16-b81d-11e6-83ea-ad95ad6db225.png

![alt_txt][K5]
[K5]:https://cloud.githubusercontent.com/assets/12142475/20815162/164b0c66-b81d-11e6-9bf8-3ff9736b4440.png

![alt_txt][K6]
[K6]:https://cloud.githubusercontent.com/assets/12142475/20815171/1c91ee8c-b81d-11e6-8f24-54eec3247f96.png



###PCA

PCAdapt in R:
```
##convert .vcf to plink 
##linux

vcftools --vcf SE.145.FINAL.recode.vcf --plink --out SE145.FINAL.plink

plink --file SE145.FINAL.plink --recode --recodeA

##R
library(pcadapt)

SE.145 <- read.pcadapt("SE145.FINAL.plink.ped", type="ped")
Summary:

        - input file      SE145.FINAL.plink.ped
        - output file     SE145.FINAL.plink.pcadapt

	- number of individuals detected:	145
	- number of loci detected:		2041

File has been sucessfully converted.

##Check the nr of PCs

x <- pcadapt(SE.145, K=20)

Reading file SE145.FINAL.plink.pcadapt...
Number of SNPs: 2041
Number of individuals: 145
Number of SNPs with minor allele frequency lower than 0.05 ignored: 53
21683 out of 295945 missing data ignored.

plot(x, option="screeplot")  ##PC for pop structure = on the steep curve
```

Choose number of PCs:

![alt_txt][PCadapt.fig1]
[PCadapt.fig1]:https://cloud.githubusercontent.com/assets/12142475/20815725/541bfc24-b81f-11e6-86a0-69105ce85742.png

Plot the PCA using population information
```
pop.SE145 
head(pop.SE145)
        indiv        pop region
1      FIN_21     X6.FIN    FIN
2      FIN_22     X6.FIN    FIN
3      FIN_23     X6.FIN    FIN
4      FIN_25     X6.FIN    FIN

poplist <- pop.SE145[,2]  ##select pops
plot(x,option="scores",pop=poplist)

poplist <- as.character(popnames[,3]) ##select regions
plot(x,option="scores",pop=poplist)
```

Populations
![alt_txt][PCadapt.Fig2]
[PCadapt.Fig2]:https://cloud.githubusercontent.com/assets/12142475/20815924/2fbd151a-b820-11e6-8d57-fb3953180da3.png


Regions
![alt_txt][PCadapt.Fig3]
[PCadapt.Fig3]:https://cloud.githubusercontent.com/assets/12142475/20815930/34fb26d4-b820-11e6-958e-ff764030d574.png



##fastStructure



##TESS3


#Env Associations

##BIOclim variables

Find the uncorrelated variables from BIOclim: 

Extract BIOclim variables. See: https://github.com/alexjvr1/Manuscripts/blob/master/6.R.temp_SE.MS1.md


```
library(corrplot)
library(caret)
datMy <- read.table("SEonly.BIOclim", header=T)
datMy.scale <- scale(datMy[1:ncol(datMy)], center=T, scale=T)
corMatMy <- cor(datMy.scale, method="spearman")
corrplot(corMatMy, order="hclust", tl.cex=1)

highlyCor <- findCorrelation(corMatMy, 0.80) #After inspection, apply correlation filter at 0.70,

#then we remove all the variable correlated with more 0.8.
datMyFiltered.scale <- datMy.scale[,-highlyCor]
corMatMy <- cor(datMyFiltered.scale)
corrplot(corMatMy, order = "hclust", tl.cex=1)

```

![alt_txt][BIOclim.all]
[BIOclim.all]:https://cloud.githubusercontent.com/assets/12142475/20832670/d86ef6c2-b88b-11e6-9ad7-e06544448f51.png

![alt_txt][BIOclim.reduced]
[BIOclim.reduced]:https://cloud.githubusercontent.com/assets/12142475/20832675/dde218c8-b88b-11e6-8158-0eabc1f40338.png





##RDA
Input files:

MAF of all loci

Geographic coordinates

Climate variables
	
	1. BIO5: Mat temp warmest month
	
	2. BIO15: Precipitation seasonality
	
	3. BIO13: Precipitation Wettest month
	
	4. BIO18: Precipitation warmest quarter (correlated with growth season)
	
	5. BIO2: Mean diurnal range

2041 loci


```
###1. MAF

#Calculate MAF for the full dataset within region using PLINK

/Users/alexjvr/2016RADAnalysis/5_SE.MS1/DEC2016_SEonly/RDA/

###Use the *nosex file to create a file for subsetting the data. Here I've made 1=within region, 2=within pop

plink --file SE145.FINAL.plink --freq --within SE145.region.cluster --out SE145.cluster

plink --file SE145.FINAL.plink --freq --within SE145.pop.cluster --out SE145.pop


###R
######Reformat PLINK output
###MAF for each locus -> melt and reformat rows as pops, and columns as loci.

SE.MAF <- read.table("SE145.pop.frq.strat", header=T)
SE.MAF2 <- SE.MAF[,c(3,2,6)]
summary(SE.MAF2)
         CLST              SNP             MAF         
 X1.Sk.Ho  : 2041   101108:35:   15   Min.   :0.00000  
 X1.Sk.SF  : 2041   101142:72:   15   1st Qu.:0.00000  
 X1.Sk.SL  : 2041   101270:3 :   15   Median :0.07143  
 X2.Upp.Gra: 2041   101367:84:   15   Mean   :0.18166  
 X2.Upp.K  : 2041   101609:30:   15   3rd Qu.:0.30000  
 X2.Upp.O  : 2041   101834:91:   15   Max.   :1.00000  
 (Other)   :18369   (Other)  :30525     

library("ggplot2")
library("reshape2")

SE.MAF3 <- melt(SE.MAF2, id.vars = c("CLST", "SNP"), variable_name = c("MAF"))
str(SE.MAF3)
head(SE.MAF3)


SE145.MAF4 <- dcast(SE.MAF3, formula= CLST ~ SNP)
head(SE145.MAF4)



##Add X infront of all locusnames. 
colnames(SE145.MAF4) <- paste("X", colnames(SE145.MAF4), sep=".")
write.csv(SE145.MAF4, file="SE.145.MAF.csv")
```


Run RDA

See this tutorial for the interpretation: REDUNDANCY ANALYSIS TUTORIAL: Landscape Genetics Paul Gugger redundancy-analysis-for-landscape-genetics.pdf on mac
```
library(vegan)

GenData <- read.csv("SE.145.MAF.csv", header=T)
GenData <- GenData[,15:2055]

Climate.Data <- read.csv("SE.145.MAF.csv", header=T)
names(Climate.Data)
Climate.variables <- Climate.Data[,1:14]
names(Climate.variables)
Climate.Data <- Climate.variables[,10:14]
Climate.Data$Lat <- Climate.variables$Lat
Climate.Data$Long <- Climate.variables$Long


##1. Run RDA with only climate data 
##H0: climate data does not affect genotype

RDA.SEfull <- rda(GenData ~ Lat + Long +bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled, Climate.Data) 
RDA.SEfull
Call: rda(formula = GenData ~ Lat + Long + bio5.scaled + bio15.scaled +
bio13.scaled + bio18.scaled + bio2.scaled, data = Climate.Data)

              Inertia Proportion Rank
Total         80.7640     1.0000     
Constrained   61.8140     0.7654    7
Unconstrained 18.9499     0.2346    7
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1   RDA2   RDA3   RDA4   RDA5   RDA6   RDA7 
25.969 13.331  7.957  4.612  4.274  3.625  2.046 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
5.153 3.555 2.913 2.256 1.959 1.732 1.381 

anova(RDA.SEfull)

Permutation test for rda under reduced model
Permutation: free
Number of permutations: 999

Model: rda(formula = GenData ~ bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled, data = Climate.Data)
         Df Variance      F Pr(>F)    
Model     5   52.894 3.4162  0.001 ***
Residual  9   27.870                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


##So H0 rejected -> climate affects genotype
##to see which variables are most important, we can plot the results in a biplot

plot(RDA.SEfull)

```

![alt_txt][RDA.SEfull]
[RDA.SEfull]:https://cloud.githubusercontent.com/assets/12142475/20836511/fbcdccf6-b89f-11e6-9845-ba7f25289f71.png



```
##Partial out geog
pRDA.geog <- rda(GenData~bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled+ Condition(Lat + Long), Climate.Data)
head(summary(pRDA.geog))

anova(pRDA.geog)
Permutation test for rda under reduced model
Permutation: free
 Number of permutations: 999

Model: rda(formula = GenData ~ bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled + Condition(Lat + Long), data = Climate.Data)
         Df Variance      F Pr(>F)    
Model     5   31.097 2.2974  0.001 ***
Residual  7   18.950       

pRDA.geog
              Inertia Proportion Rank
Total         80.7640     1.0000     
Conditional   30.7168     0.3803    2
Constrained   31.0972     0.3850    5
Unconstrained 18.9499     0.2346    7
Inertia is variance 

Eigenvalues for constrained axes:
  RDA1   RDA2   RDA3   RDA4   RDA5 
13.056  7.441  4.735  3.640  2.226 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
5.153 3.555 2.913 2.256 1.959 1.732 1.381 

plot(pRDA.geog, main="pRDA.geog")
```

![alt_txt][pRDA.geog]
[pRDA.geog]:https://cloud.githubusercontent.com/assets/12142475/20835874/21af979a-b89d-11e6-8864-679bc492e3a9.png


```
##Partial out climate
pRDA.climate <- rda(GenData~Lat+Long + Condition(bio5.scaled + bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled), Climate.Data)
pRDA.climate
Call: rda(formula = GenData ~ Lat + Long + Condition(bio5.scaled +
bio15.scaled + bio13.scaled + bio18.scaled + bio2.scaled), data =
Climate.Data)

              Inertia Proportion Rank
Total         80.7640     1.0000     
Conditional   52.8942     0.6549    5
Constrained    8.9198     0.1104    2
Unconstrained 18.9499     0.2346    7
Inertia is variance 

Eigenvalues for constrained axes:
 RDA1  RDA2 
5.880 3.039 

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7 
5.153 3.555 2.913 2.256 1.959 1.732 1.381 

head(summary(pRDA.climate))


```

![alt_txt][pRDA.climate]
[pRDA.climate]:https://cloud.githubusercontent.com/assets/12142475/20835977/9ee485cc-b89d-11e6-80e3-278ef21e4f4f.png


