#Adaptation to similar environmental gradients at very different spatial scales

Important papers: 

1. Rellstab et al. 2015 A practical guide to Environmental association analysis in landscape genomics
2. Cano et al. 2011. First Generation linkage map for the common frog (for recombination rates)
3. Lottheros & Whitlock 2014: Evaluation of demographic history and neutral paramaterization on the performance of Fst outlier analyses
4. Yeaman & Guillaume 2009
5. Kopp & Matuszewski 2013: Review looks useful for looking at modelling approaches under different selection scenarios. 
6. Bridle et al. 2010: What prevents adaptation at range margins? Individual simulations
7. Genome scan methods against more complex models: when and how much should we trust them? (2015 - I think) de Villemereuil et al. 


- What is the genomic signature of adaptation to the same environmental gradient?
- When can we expect parallel evolution? 
      - spatial scale
      - gene flow
      - standing genetic variation
      - trait complexity
      - strength of selection


Simulations??


Important papers:
1. ###The consequences of gene flow for local adaptation and differentiation: a two-locus two-deme model
Akerman & BÃ¼rger 2012
"We consider a population subdivided into two demes connected by migration
in which selection acts in opposite direction.We explore the effects of recombination
and migration on the maintenance of multilocus polymorphism, on local adaptation,
and on differentiation by employing a deterministic model with genic selection on
two linked diallelic loci (i.e., no dominance or epistasis)."


Local adaptation helps to maintain ecologically/biologically relevant variation (Whitlock 2015 Am Nat). 

Yeaman 2015 suggests that signatures of selection will be difficult to find when multiple small effect loci are involved: both because the effect size is small, and becuase the genome scan would not neccessarily include all the relevant loci. 

Berg & Coop (2014) suggest a method to detect the cumulative effect of these loci. But need prior information (GWAS). 

**New Fst oulier analysis method that reduces false positives without much loss of power (OutFLANK; Lotheros & Whitlock 2015). 


**Mol Ecol Dec 2015 - Special issue on genome scans. 
- Many things can produce similar patterns to hitchhiking models, thus produce a high false positive rate
      - population size changes (e.g. Teshima et al. 2006)
      - Population structure (Excoffier et al. 2009)
      - Background selection (Stephan 2010)
- Low power of detection 
      - heterogeneous mutation and recombination rate (Roesti et al. 2012 & Tine et al. 2014)
      - past demography (even when corrected for) (Bierne et al. 2011)
      - Genetic architecture of the selected trait (Le Corre & Kremer 2012)
- one of the problems is that we need an appropriate model for the expectation of neutral evolution. So it is quite important to know the demographic history of a system before any inferences are made. 
- It also seems like background selection (i.e. selection vs deletarious mutations) cannot be ignored, and may in fact 


SNP validation: 

1. Check the frequency of SNPs above a certain threshold. (SNPs in sequnces >XXbp were much higher than expected). 
